---
title: "Start codon context for Candida Albicans"
author: Edward Wallace, ewjwallace@gmail.com
date: 12 Oct 2018
output:
  # html_notebook: default
  html_document:
    toc: true
    toc_depth: 4
---

# Summary

This presents the analysis of Candida Albicans SC5314 start codon usage and context. This uses the C. albicans translation estimates from Nuzzey 2014.

### Load Packages

```{r setup,message=FALSE,echo=FALSE,cache=FALSE}
knitr::opts_chunk$set(warning=FALSE,message=FALSE,echo=FALSE,cache=TRUE,
                      results="hide",
                      fig.path="figure/CalbicansATGContext-",
                      cache.path="cache/CalbicansATGContext-")
# load common functions
source("../SharedScripts/sharedScripts.R")
# setwd("~/Repos/CryptoTranscriptome2018/ATGContextOthers/")
```


# Here we go

## Expression: RNA abundance and ribosome-protected-fragments

### Load expression data

```{r load_ribosum,results="show"}

tpms <- read_tsv("data_in/ExpressionTranslation/CalbicansNuzzey2014_tpms.tsv") %>%
    dplyr::rename(Gene=ORF,RPF=tpm)


tpms %>%
    arrange(desc(RPF)) 

ribosum <- tpms %>% select(Gene,RPF)

hiTrans <- ribosum %>%
    arrange(desc(RPF)) %>%
    head(n=300) %>%
    .$Gene

midTrans <- ribosum %>%
    arrange(desc(RPF)) %>%
    head(n=3000) %>%
    .$Gene

nonterribleTrans <- ribosum %>%
    arrange(desc(RPF)) %>%
    head(n=4500) %>%
    .$Gene

# enoughRNA <- ribosum %>%
#     arrange(desc(RNA)) %>%
#     head(n=5000) %>%
#     .$Gene
```

### Ribosome occupancy mostly tracks RNA abundance

```{r plot_TE,dependson="load_ribosum",fig.height=3.9,fig.width=4.1,eval=FALSE}
TE_unusual <- 
    ribosum %>%
    filter( (RPF > 1.8e4) |
                (RPF > 100 & RPF > RNA * 5 ) | 
                (RNA > 150 & RPF < RNA / 10 ))


ggplot(data=ribosum,aes(x=RNA,y=RPF)) +
    geom_diagline() +
    geom_point(size=.8,colour="grey50") +
    geom_point(data=TE_unusual,aes(colour=Gene),size=1) + 
    geom_text_repel(data=TE_unusual,aes(label=Gene,colour=Gene)) +
    scale_loglog(limits=c(1e-1,2e4),
                  oob=scales::squish,
                  expand=c(0.01,0.01)) +
    labs(x="RNA abundance (TPM of aORF)",
         y="Ribosome occupancy (TPM of aORF)") +
    theme(legend.position="none")
```

## ATG Context

### Load context data

```{r load_context,results="show"}
ctbl_big <- read_ATGcontext_table("data_in/CAlbi_allATGcontext.table.txt")
ctbl_big
```

### Annotated ATGs have a Kozak consensus sequence

```{r aATG_context,fig.width=4,fig.height=1}
ggseqlogo(data=ctbl_big$aATG.context,namespace="TCAG") + 
    theme_nothing()
```

### Highly translated Annotated ATGs have a Kozak consensus sequence

That's for hiTrans, the top 4% (400) translated genes by RPF TPM.

```{r a_hi_context,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big,Gene %in% hiTrans)$aATG.context,
                      namespace="TCAG") + 
    theme_nothing()
```

### Cytoplasmic Ribosome Annotated ATGs 



```{r cytoribo_context,dependson="load_context",fig.width=4,fig.height=1}
cytoribolist <- read_tsv("data_out/ComplexOrthologs/cytoribosome_PTRcurated6.txt",
                        comment="#") %>%
    filter(SPECIES=="CANAL") %>%
    .$GENE_ID
ggseqlogo(data=filter(ctbl_big,Gene %in% cytoribolist)$aATG.context,
                      namespace="TCAG") + 
    theme_nothing()
```


### Downstream ATGs don't have a consensus

First downstream ATG

```{r d1_context,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big,nchar(d1.context) ==24)$d1.context,
                      namespace="TCAG") + 
    theme_nothing()
```

### Downstream ATGs in frame and highly translated don't have a consensus

Except for 3rd-codon-position bias.

```{r d1_hi_frame0_context,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big,
                      nchar(d1.context) == 24,
                      d1.frame==0,
                      Gene %in% hiTrans)$d1.context,
                      namespace="TCAG") + 
    theme_nothing()
```

## Calculate Information content and scores of consensus motif

### Calculate a wide and a narrow consensus sequence

Calculate motif score against the position weight matrix (pwm) for both **narrow** (-5 from ATG through to ATG) and **wide** (-9 from ATG to +3) kozak consensus motif. These motifs are taken from the top 5% highly translated genes.

```{r kozak_consensus,fig.width=4,fig.height=1}
kozak_all_PFM <- filter(ctbl_big,Gene %in% hiTrans) %>%
    .$aATG.context %>%
    str_to_upper() %>%
    consensusMatrix() 

kozak_w_PWM <- PWM(kozak_all_PFM[,3:15])
kozak_n_PWM <- PWM(kozak_all_PFM[,9:15])
```


```{r write_PFM, dependson="calc_scores",results="show"}
cat(
    "# PFM_hiTrans_Ca.txt
# Position Frequency Matrix for aATGs of 400 highest-translated genes from C. albicans SC5314
# This is of the wide ATG context NNNNNNNNNNNNATGNNNNNNNNN
# Column is position, Row is nt count in order: ACGT.
#
# Edward Wallace, June 2018
",
file="data_out/PFM_hiTrans_Ca.txt")
kozak_all_PFM %>%
    as.tibble() %>%
    write_tsv("data_out/PFM_hiTrans_Ca.txt",
              append = TRUE,col_names=FALSE)
```

### Estimate the information content 

Using the sequence logo, details on https://en.wikipedia.org/wiki/Sequence_logo

```{r kozak_inf,dependson=c("kozak_consensus","cytoribo_context"),fig.width=4,fig.height=1,results="show"}
inf_content_narrow <- 
    tribble(
        ~Genes, ~ATG, ~Width, ~Infon,
        "All", "aATG", "narrow", 
        ctbl_big %>%
            .$aATG.context %>%
            infonfromseqs(sstart=9L,send=12L),
        "HiTrans", "aATG", "narrow", 
        ctbl_big %>%
            filter(Gene %in% hiTrans) %>%
            .$aATG.context %>%
            infonfromseqs(sstart=9L,send=12L),
        "CytoRibo", "aATG", "narrow",
        ctbl_big %>%
            filter(Gene %in% cytoribolist) %>%
            .$aATG.context %>%
            infonfromseqs(sstart=9L,send=12L),
        "All", "d1ATG", "narrow",
        ctbl_big %>%
            .$d1.context %>%
            na.omit() %>%
            infonfromseqs(sstart=9L,send=12L),
        "HiTrans", "d1ATG", "narrow", 
        ctbl_big %>%
            filter(Gene %in% hiTrans) %>% 
            .$d1.context %>%
            na.omit() %>%
            infonfromseqs(sstart=9L,send=12L),
        "CytoRibo", "d1ATG", "narrow",
        ctbl_big %>%
            filter(Gene %in% cytoribolist) %>%
            .$d1.context %>%
            na.omit() %>%
            infonfromseqs(sstart=9L,send=12L)
    )

write_tsv(inf_content_narrow %>%
              mutate_if(is.numeric,round3),
          "data_out/inf_content_narrow_Ca.txt")
inf_content_narrow
```

Information content in bits of highly-translated consensus (excluding 6 bits from ATG), **narrow** is `r infontot(kozak_all_PFM[,5:9]) %>% round(2)`, of **wide** is `r infontot(kozak_all_PFM[,-(10:12)]) %>% round(2)`.


### Estimate information content per base across start context

```{r kozak_inf_perbase,dependson=c("kozak_consensus","cytoribo_context"),fig.width=6,fig.height=1.4,results="show"}
inf_content_perbase <- 
    tibble(i = 1:24, 
           Pos = c(-12:-1, 1:12),
        All = ctbl_big %>%
            .$aATG.context %>%
            infonfromseqs(scope="all"),
        HiTrans = ctbl_big %>%
            filter(Gene %in% hiTrans) %>% 
            .$aATG.context %>%
            infonfromseqs(scope="all"),
        CytoRibo = ctbl_big %>%
            filter(Gene %in% cytoribolist) %>% 
            .$aATG.context %>%
            infonfromseqs(scope="all")
    )

write_tsv(inf_content_perbase %>%
              mutate_if(is.numeric,round3),
          "data_out/inf_content_perbase_Ca.txt")
# inf_content_perbase

plot_inf_perbase(inf_content_perbase)
```

### Calculate scores of aATG, dATG, uATG against Kozak consensus

```{r calc_scores, dependson="kozak_consensus"}
scores <- 
    ctbl_big %>%
    transmute(Gene,
              aATG.scorekn = PWMscoren(str_to_upper(aATG.context),kozak_n_PWM),
              d1.scorekn   = PWMscoren(d1.context,kozak_n_PWM),
              # u1.scorekn   = PWMscoren(u1.context,,kozak_n_PWM)
              NULL
    ) %>%
    mutate(d1vsan = d1.scorekn - aATG.scorekn,
           # u1vsan = u1.scorekn - aATG.scorekn,
           NULL) 

# ribotbl_scoresenough <-
#     ribosum %>%
#     filter(Gene %in% enoughRNA) %>%
#     # left_join(ribotbl,by="Gene") %>% 
#     left_join(scores,by="Gene")

```

We calculate scores using Biostrings::PWMscoreStartingAt.

The best description I could find of this method is: https://support.bioconductor.org/p/61520/

It is just the sum of the matrix product of the PWM with the sequence.


Write scores to file *scores_kozak_Ca.txt*.

```{r write_scores, dependson="calc_scores",results="show"}
scores
cat(
    "# scores_kozak_Ca.txt
# Scores of ATG context against Kozak sequence, Candida albicans
# Kozak sequence derived from 400 highest-translated genes by RPF
#
# Gene: Ca systematic gene name
# aATG.scorekn: score of annotated ATG against narrow Kozak sequence (NNNNATG)
# d1.scorekn: score of 1st downstream ATG against narrow
# xx.scorekw: analogous scores against wide Kozak sequence (NNNNNNNNNNATG)
#
# Edward Wallace, Oct 2018
",
file="data_out/scores_kozak_Ca.txt")
scores %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/scores_kozak_Ca.txt",
              append = TRUE,col_names=TRUE)
```

### Plot against narrow consensus (-5 to ATG)

```{r plot_scores_n, dependson="calc_scores", fig.width=5,fig.height=1.8}

sbandwidthn = 0.025

ggplot(data=scores) +
    geom_density(aes(x=aATG.scorekn,colour="aATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    geom_density(aes(x=d1.scorekn,colour="dATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    # geom_density(aes(x=u1.scorekn,colour="uATG"),
    #             kernel="rectangular",bw=sbandwidthn) +
    geom_density(data=filter(scores,Gene %in% hiTrans),
                 aes(x=aATG.scorekn,colour="aATG, top 5%"),
                 kernel="rectangular",bw=sbandwidthn) +
    scale_colour_manual(values=c("aATG"="dodgerblue",
                                 "aATG, top 5%"="darkblue",
                                 # "uATG"="red",
                                 "dATG"="green3") ) +
    labs(x="ATG Kozak score, Ca") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank(),
          legend.title=element_blank())
```


## Mutual information of different positions around ATG

### For all annotated genes

```{r mutinf,dependson=c("load_context","load_riboviz"),results="show",fig.height=4,fig.width=5}
mutinf <- mutinfostr(ctbl_big$aATG.context)
# mutinf

plot_mutinfo(mutinf)
```

### For highly translated genes

```{r mutinf_hiTrans,dependson=c("load_context","load_riboviz"),results="show",fig.height=4,fig.width=5}
ctbl_big %>%
    filter(Gene %in% hiTrans) %>%
    .$aATG.context %>%
    mutinfostr %>%
    plot_mutinfo
```


### For cytoplasmic ribosomal proteins

```{r mutinf_cytoribo,dependson=c("load_context","load_riboviz"),results="show",fig.height=4,fig.width=5}
ctbl_big %>%
    filter(Gene %in% cytoribolist) %>%
    .$aATG.context %>%
    mutinfostr %>%
    plot_mutinfo

```


### What is the -2 correlation in cyto ribo genes?

```{r cytoribo_contextm2,dependson="load_context",fig.width=4,fig.height=3}

seqlogoFilterGenePos <- function(datain=ctbl_big,Genelist=hiTrans,
                                 pos=8L, nt="T") {
    datafiltered <- filter(datain,
                          Gene %in% Genelist,
                          str_sub(aATG.context,start=pos,end=pos)==nt
    )
    ggseqlogo(data=datafiltered$aATG.context,
    namespace="TCAG") + 
        theme_nothing() +
        annotate(geom="text",x=21,y=1,
                 label=paste0("n = ", nrow(datafiltered)))
}

plot_grid(seqlogoFilterGenePos(Genelist=cytoribolist,pos=11L,nt="T"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=11L,nt="C"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=11L,nt="A"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=11L,nt="G"),
    ncol=1
)
    
```

There are definitely correlations there.


### What about the -4 correlation in cyto ribo genes?

```{r cytoribo_contextm4,dependson="load_context",fig.width=4,fig.height=3}

seqlogoFilterGenePos <- function(datain=ctbl_big,Genelist=hiTrans,
                                 pos=8L, nt="T") {
    datafiltered <- filter(datain,
                          Gene %in% Genelist,
                          str_sub(aATG.context,start=pos,end=pos)==nt
    )
    ggseqlogo(data=datafiltered$aATG.context,
    namespace="TCAG") + 
        theme_nothing() +
        annotate(geom="text",x=21,y=1,
                 label=paste0("n = ", nrow(datafiltered)))
}

plot_grid(seqlogoFilterGenePos(Genelist=cytoribolist,pos=9L,nt="T"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=9L,nt="C"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=9L,nt="A"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=9L,nt="G"),
    ncol=1
)
    
```

Weaker there.


## Compare aATG and dATG context by gene

### Most dATG scores are less than aATG scores

```{r dvsaATG_scoresdens, dependson="calc_scores", fig.width=3.5,fig.height=1.8}

ggplot(data=scores) +
    geom_vline(xintercept=0,size=0.3) +
    geom_density(aes(x=d1vsan,colour="narrow"),kernel="rectangular") +
    # geom_density(aes(x=d1vsaw,colour="wide"),kernel="rectangular")  +
    labs(x="difference in score, dATG vs aATG",
         colour="context") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank(),
          legend.position=c(0.8,0.8))
```

### For highly translated genes, most dATG scores are much less than aATG

```{r dvsaATG_scoresn, dependson="calc_scores", fig.width=3.5,fig.height=3.5}
highdiffn_dvsaATG <- scores %>%
    arrange(desc(d1vsan)) %>%
    head(n=400) %>%
    select(Gene,aATG.scorekn,d1.scorekn)

ggplot(data=scores %>% filter(!is.na(d1.scorekn)),
       aes(x=aATG.scorekn,y=d1.scorekn)) +
    geom_diagline() +
    geom_point(size=0.5,colour="grey50") +
    geom_point(data=highdiffn_dvsaATG,
               colour="red",size=1) +
    geom_point(data=scores %>% 
                   filter(Gene %in% hiTrans),
               colour="blue",size=1) +
    geom_point(data=highdiffn_dvsaATG %>% 
                   filter(Gene %in% hiTrans),
               colour="purple4",size=1.3) +
    labs(x="aATG score, narrow",
         y="dATG score, narrow")
```

Red: high dATG vs aATG **Kozak** score.
Blue: highly translated.
Purple: both.

### Small negative correlation between dATG and aATG score

R = `r with(scores, cor(aATG.scorekn,d1.scorekn,use="pairwise.complete.obs")) %>% round3`

## Genes with unusual dATG vs aATG narrow score

Those genes are in this list:

```{r highdiffn_dvsaATG,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG
```


Some of these are well translated:

```{r highdiffn_dvsaATG_hiTrans,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene, d1.frame,d1.posATG) ) %>% 
    filter(Gene %in% hiTrans) 
```

### dATG in frame with ATG

Files with high difference in narrow score, filtered for reasonable amounts of translation, in frame. We did **not** filter for top 50% of RNA because we did not directly quantify RNa levels here.

Saved to *dvsaATG_highdiffn_inframe.txt*.

```{r highdiffn_dvsaATG_inframe,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG_enoughinframe <- 
    highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene,d1.frame) ) %>%
    filter(Gene %in% nonterribleTrans, d1.frame==0) %>%
    select(Gene,aATG.scorekn,d1.scorekn)

highdiffn_dvsaATG_enoughinframe

cat(
    "# dvsaATG_highdiffn_inframe_Ca.txt
# Genes with in-frame high-score dATGs
# In C. albicans SC5314:
#   - in top 5% by difference, aATG.scorekn - d1.scorekn
#   - in top 75% by translation
#   - dATG is in frame with aATG
#
# Gene: systematic gene name
# aATG.scorekn: annotated ATG score against narrow Kozak consensus sequence, strain xx
# d1.scorekn: 1st downstream ATG score against narrow Kozak consensus sequence, strain xx
#
# Edward Wallace, Dec 2018
",
file="data_out/dvsaATG_highdiffn_inframe_Ca.txt")
highdiffn_dvsaATG_enoughinframe %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/dvsaATG_highdiffn_inframe_Ca.txt",append = TRUE,col_names=TRUE)
```

- C5_00840W_A, RNA transcription and export complex,  TEX1 / YNL253W ortholog

See more below.

### dATG out of frame with ATG

Files with high difference in narrow score, filtered for reasonable amounts of translation, out of frame. 

Saved to *dvsaATG_highdiffn_outframe_Ca.txt*.

```{r highdiffn_dvsaATG_outframe,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG_enoughoutframe <- 
    highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene,d1.frame) ) %>%
    filter(Gene %in% nonterribleTrans, d1.frame!=0) %>%
    select(Gene,aATG.scorekn,d1.scorekn)

highdiffn_dvsaATG_enoughoutframe

cat(
    "# dvsaATG_highdiffn_outframe_Ca.txt
# Genes with in-frame high-score dATGs
# In C. albicans SC5314:
#   - in top 5% by difference, aATG.scorekn - d1.scorekn
#   - in top 75% by translation
#   - dATG is out-of-frame with aATG
#
# Gene: systematic gene name
# aATG.scorekn: annotated ATG score against narrow Kozak consensus sequence, strain xx
# d1.scorekn: 1st downstream ATG score against narrow Kozak consensus sequence, strain xx
#
# Edward Wallace, Dec 2018
",
file="data_out/dvsaATG_highdiffn_outframe_Ca.txt")
highdiffn_dvsaATG_enoughoutframe %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/dvsaATG_highdiffn_outframe_Ca.txt",append = TRUE,col_names=TRUE)
```


## Compare score difference to localization predictions


### Load predictions from mitofates

In input file *CAlbi_mitofates.txt*.

```{r load_mitofates}
mitofates <- read_mitofates("data_in/Calbi_mitofates.txt")

mitocounts <- 
    mitofates %>% 
    group_by(Pred_preseq) %>% 
    count() 
mitobreaks <- mitocounts %>%
    mutate(label=paste0(Pred_preseq," (n = ",n,")")) %>%
    .$label %>%
    set_names(mitocounts$Pred_preseq)

mitopredlist <- mitofates %>%
    filter(Pred_preseq =="Yes") %>%
    .$Gene
```


### Ca Genes with high dATG vs aATG score might be enriched in mitochondrial presequences?

```{r dvsaATG_scoresdiffn_mitofates, dependson=c("calc_scores","load_mitofates"), fig.width=5,fig.height=1.8}

ggplot(data=scores %>% 
           filter(!is.na(d1.scorekn)) %>%
           left_join( mitofates %>% 
                          select(Gene,Prob_preseq,Pred_preseq)
                      ) ,
       aes(x=d1.scorekn-aATG.scorekn,colour=Pred_preseq)) +
    geom_vline(xintercept=0,size=0.3) +
    geom_density(kernel="rectangular",bw=0.025) +
    scale_colour_manual(values=c("Yes"="magenta","No"="grey50"),
                        labels=mitobreaks)  + 
    labs(x="Kozak score, difference dATG - aATG", colour="Mito. pred") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank())
```

### Ca mito-localized genes also do not have a distinctive aATG context

Although the +5 T is striking.

```{r aATG_motif_mitofates, dependson=c("calc_scores","load_mitofates"), fig.width=4,fig.height=1}

ggseqlogo(data=ctbl_big %>%
              filter(Gene %in% mitopredlist) %>%
              .$aATG.context,
          namespace="TCAG")  + 
    theme_nothing()

```

## Genes with unusual dATG vs aATG narrow score AND predicted mito loc

Those genes are in this list:

```{r highdiffn_dvsaATG_mitopred,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene, d1.frame,d1.posATG) ) %>%
    left_join( ribosum ) %>%
    filter(Gene %in% mitopredlist) %>%
    print(n=26)
```

Only some have in-frame ATG.

* C2_06190W_A / ILV1Putative threonine dehydratase; regulated by *Gcn4* and Gcn2; induced by amino acid starvation (3-AT);


### Some more decently-translated ones with in-frame aATG and mito. loc

```{r highdiffn_dvsaATG_mitopred_hiTrans,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene, d1.frame,d1.posATG) ) %>%
    left_join( ribosum ) %>%
    filter(Gene %in% mitopredlist, Gene %in% midTrans, d1.frame==0) 
```

* C2_09260C_A GLO2 hydroxyacylglutathione hydrolase, Scer has GLO2 (cyto) and GLO4 (mito) paralogs
* C1_00490C_A TTR1 Putative glutaredoxin, regulated by Gcn2 and Gcn4, Scer has GRX1 (cyto/nuc) and GRX2 (mito/cyto) paralogs
* C3_02270W_A conserved hypothetical protein with signal peptide, Scer MCO8 is found in mitochondria.
* C6_02410W_A, NAD(+) diphosphatase homolog, Scer NPY1 / YGL067W localizes to peroxisome and cytoplasm.
* C1_08060W_A OFR1 Protein involved in regulation of white-opaque switching
* CR_04300W_A tRNA (cytosine-5-)-methyltransferase, Scer homolog NCL1 is nuclear.
* CR_04590C_A PDX3 Pyridoxamine-phosphate oxidase
* C6_03720W_A ALA1 Alanyl-tRNA synthetase; translational regulation generates cytoplasmic and mitochondrial forms.
* C5_01520C_A GLR1 Glutathione reductase
* C2_10270W_A PUS7 Pseudouridine synthase
* C1_01040W_A Methionine-R-sulfoxide reductase, Scer homolog MXR2 is mitochondrial.
* CR_01550C_A 3'--> 5' exonuclease and endonuclease
* C6_03820C_A conserved protein of unknown function with homology to probable FAD synthase and Molybdenum cofactor biosynth proteins??
* C2_06640C_A tRNA-Val synthetase VAS1
* C6_00390W_A tRNA-Serine synthetase, there is another paralog C3_02780W_A, homology to cyto SES1, and a mito DIA4 homolog.


Several of these are conserved (ALA1, VAS1, PUS7, GLOs)!






<a href="#top">Back to table of contents</a>
