---
title: "Start codon context for Candida Albicans"
author: Edward Wallace, ewjwallace@gmail.com
date: 7 Oct 2018
output:
  # html_notebook: default
  html_document:
    toc: true
    toc_depth: 4
---

# Summary

This presents the analysis of Candida Albicans SC5314 start codon usage and context. This uses the C. albicans translation estimates from Nuzzey 2014.

### Load Packages

```{r setup,message=FALSE,echo=FALSE,cache=FALSE}
knitr::opts_chunk$set(warning=FALSE,message=FALSE,echo=FALSE,cache=TRUE,
                      results="hide",
                      fig.path="figure/CalbicansATGContext-",
                      cache.path="cache/CalbicansATGContext-")
library(tidyverse)
library(cowplot)
library(ggrepel)
library(ggseqlogo)
library(Biostrings)

theme_set(theme_cowplot(font_size=12) + 
              theme(strip.background = element_blank()) )



geom_diagline <- function(linetype='solid',size=0.1,colour="grey20",...) {
    geom_abline(slope=1,intercept=0,linetype=linetype,colour=colour)
}

scientific_10 <- function(x) {
    xout <- gsub("1e", "10^{", format(x),fixed=TRUE)
    xout <- gsub("{-0", "{-", xout,fixed=TRUE)
    xout <- gsub("{+", "{", xout,fixed=TRUE)
    xout <- gsub("{0", "{", xout,fixed=TRUE)
    xout <- paste(xout,"}",sep="")
    return(parse(text=xout))
}

scale_x_log10nice <- function(name=waiver(),omag=seq(-10,20),...) {
    breaks10 <- 10^omag
    scale_x_log10(name,breaks=breaks10,labels=scientific_10(breaks10),...)
}

scale_y_log10nice <- function(name=waiver(),omag=seq(-10,20),...) {
    breaks10 <- 10^omag
    scale_y_log10(name,breaks=breaks10,labels=scientific_10(breaks10),...)
}

scale_loglog <- function(...) {
    list(scale_x_log10nice(...),scale_y_log10nice(...))
}

round3 <- function(x) round(x,digits=3)

logmean <- function(x) {
    lx <- log(x)
    exp( mean( lx[is.finite(lx)] ) )
}
# setwd("~/Repos/CryptoTranscriptome2018/ATGContextOthers/")
```


# Here we go

## Expression: RNA abundance and ribosome-protected-fragments

### Load expression data

```{r load_ribosum_Ca,results="show"}

tpms_Ca <- read_tsv("data_in/ExpressionTranslation/CalbicansNuzzey2014_tpms.tsv") %>%
    dplyr::rename(Gene=ORF,RPF=tpm)


tpms_Ca %>%
    arrange(desc(RPF)) 

ribosum_Ca <- tpms_Ca %>% select(Gene,RPF)

hiTrans_Ca <- ribosum_Ca %>%
    arrange(desc(RPF)) %>%
    head(n=300) %>%
    .$Gene

midTrans_Ca <- ribosum_Ca %>%
    arrange(desc(RPF)) %>%
    head(n=3000) %>%
    .$Gene

# enoughRNA_Ca <- ribosum_Ca %>%
#     arrange(desc(RNA)) %>%
#     head(n=5000) %>%
#     .$Gene
```

### Ribosome occupancy mostly tracks RNA abundance

```{r plot_TE_Ca,dependson="load_ribosum_Ca",fig.height=3.9,fig.width=4.1,eval=FALSE}
TE_unusual_Ca <- 
    ribosum_Ca %>%
    filter( (RPF > 1.8e4) |
                (RPF > 100 & RPF > RNA * 5 ) | 
                (RNA > 150 & RPF < RNA / 10 ))


ggplot(data=ribosum_Ca,aes(x=RNA,y=RPF)) +
    geom_diagline() +
    geom_point(size=.8,colour="grey50") +
    geom_point(data=TE_unusual_Ca,aes(colour=Gene),size=1) + 
    geom_text_repel(data=TE_unusual_Ca,aes(label=Gene,colour=Gene)) +
    scale_loglog(limits=c(1e-1,2e4),
                  oob=scales::squish,
                  expand=c(0.01,0.01)) +
    labs(x="RNA abundance (TPM of aORF)",
         y="Ribosome occupancy (TPM of aORF)") +
    theme(legend.position="none")
```

## ATG Context

### Load context data

```{r load_context_Ca,results="show"}
ctbl_big_Ca <- read_tsv("data_in/CAlbi_allATGcontext.table.txt",
                        skip=1,
                        col_names=c("gene","aATG.context","aATG.pos",
                                    "d1.context","d1.posTSS","d1.posATG","d1.frame")
                        ) %>% 
    dplyr::rename(Gene=gene) %>%
    filter(!is.na(aATG.context) ) %>%
    mutate(aATG.context=toupper(aATG.context))
ctbl_big_Ca
```

### Annotated ATGs have a Kozak consensus sequence

```{r aATG_context_Ca,fig.width=4,fig.height=1}
ggseqlogo(data=ctbl_big_Ca$aATG.context,namespace="TCAG") + 
    theme_nothing()
```

### Highly translated Annotated ATGs have a Kozak consensus sequence

That's for hiTrans_Ca, the top 4% (400) translated genes by RPF TPM.

```{r a_hi_context_Ca,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big_Ca,Gene %in% hiTrans_Ca)$aATG.context,
                      namespace="TCAG") + 
    theme_nothing()
```



### Downstream ATGs don't have a consensus

First downstream ATG

```{r d1_context_Ca,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big_Ca,nchar(d1.context) ==24)$d1.context,
                      namespace="TCAG") + 
    theme_nothing()
```

### Downstream ATGs in frame and highly translated don't have a consensus

Except for 3rd-codon-position bias.

```{r d1_hi_frame0_context_Ca,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big_Ca,
                      nchar(d1.context) == 24,
                      d1.frame==0,
                      Gene %in% hiTrans_Ca)$d1.context,
                      namespace="TCAG") + 
    theme_nothing()
```

## Calculate Information content and scores of consensus motif

### Calculate a wide and a narrow consensus sequence

Calculate motif score against the position weight matrix (pwm) for both **narrow** (-5 from ATG through to ATG) and **wide** (-9 from ATG to +3) kozak consensus motif. These motifs are taken from the top 5% highly translated genes.

```{r kozak_consensus_Ca,fig.width=4,fig.height=1}
kozak_w_PFM_Ca <- filter(ctbl_big_Ca,Gene %in% hiTrans_Ca) %>%
    .$aATG.context %>%
    str_sub(start=4L,end=18L) %>%
    str_to_upper() %>%
    consensusMatrix() 

kozak_w_PWM_Ca <- PWM(kozak_w_PFM_Ca)
kozak_n_PWM_Ca <- kozak_w_PFM_Ca[,5:12] %>% PWM()
```


```{r write_PFM_Ca, dependson="calc_scores_Ca",results="show"}
cat(
    "# PFM_hiTrans_Ca.txt
# Position Frequency Matrix for aATGs of 400 highest-translated genes from Sc
# This is of the wide ATG context NNNNNNNNNATGNNN
# Column is position, Row is nt count in order: ACGT.
#
# Edward Wallace, June 2018
",
file="data_out/PFM_hiTrans_Ca.txt")
kozak_w_PFM_Ca %>%
    as.tibble() %>%
    write_tsv("data_out/PFM_hiTrans_Ca.txt",
              append = TRUE,col_names=FALSE)
```

### Estimate the information content 

Using the sequence logo, details on https://en.wikipedia.org/wiki/Sequence_logo

```{r kozak_inf_Ca,fig.width=4,fig.height=1}
entropy1 <- function(counts,ssize=4,pseudocount=0.5) {
    counts <- counts + pseudocount
    ntot = sum(counts)
    freqs = counts / sum(counts)
    Hentropy = - sum( freqs * log2(freqs) ) 
    ecorrection = ( ssize - 1 ) / ( log(2) *  ntot )
    information = log2(ssize) - Hentropy
    return(information)
}

entropytot <- function(pfm,ssize=4,pseudocount=0.5) {
    apply(pfm,2,entropy1) %>% sum()
}

entropytot(kozak_w_PFM_Ca)
# entropytot(kozak_w_PFM_Ca[,5:12])
entropytot(kozak_w_PFM_Ca[,-(10:12)])
entropytot(kozak_w_PFM_Ca[,5:9])
```

Information content in bits of highly-translated consensus (excluding 6 bits from ATG), **narrow** is `r entropytot(kozak_w_PFM_Ca[,5:9]) %>% round(2)`, of **wide** is `r entropytot(kozak_w_PFM_Ca[,-(10:12)]) %>% round(2)`.

### Calculate scores of aATG, dATG, uATG against Kozak consensus

```{r calc_scores_Ca, dependson="kozak_consensus_Ca"}

PWMscore <- function(seqs,pwm,startl) {
    # calculate scores for a character vector seqs against
    # pwm from given starting position
    # after checking strings are non-missing and start with ACTG
    vapply(str_sub(seqs,start=startl),
           function(ss) {
               if ( is.na(ss) ) {
                   return(NA)
               } else if ( str_detect(ss,"\\A[ACTG]") ) {
                   return( PWMscoreStartingAt(pwm,ss) )
               } else {
                   return(NA) 
               } 
           } ,
           FUN.VALUE = 0)
}

PWMscoren <- function(seqs,pwm=kozak_n_PWM_Ca,startl=8) {
    PWMscore(seqs,pwm,startl) 
}

PWMscorew <- function(seqs,pwm=kozak_w_PWM_Ca,startl=4) {
    PWMscore(seqs,pwm,startl) 
}

is.context <- function(x) str_detect(x,"context")

scores_Ca <- 
    ctbl_big_Ca %>%
    transmute(Gene,
              aATG.scorekn = PWMscoren(str_to_upper(aATG.context)),
              d1.scorekn   = PWMscoren(d1.context),
              # u1.scorekn   = PWMscoren(u1.context),
              # aATG.scorekw = PWMscorew(str_to_upper(aATG.context)),
              # d1.scorekw   = PWMscorew(d1.context),
              # u1.scorekw   = PWMscorew(u1.context),
              NULL
    ) %>%
    mutate(d1vsan = d1.scorekn - aATG.scorekn,
           # u1vsan = u1.scorekn - aATG.scorekn,
           # d1vsaw = d1.scorekw - aATG.scorekw,
           # u1vsaw = u1.scorekw - aATG.scorekw,
           NULL) 

# ribotbl_scoresenough_Ca <-
#     ribosum_Ca %>%
#     filter(Gene %in% enoughRNA_Ca) %>%
#     # left_join(ribotbl_Ca,by="Gene") %>% 
#     left_join(scores_Ca,by="Gene")

```

We calculate scores using Biostrings::PWMscoreStartingAt.

The best description I could find of this method is: https://support.bioconductor.org/p/61520/

It is just the sum of the matrix product of the PWM with the sequence.


Write scores to file *scores_kozak_Ca.txt*.

```{r write_scores_Ca, dependson="calc_scores_Ca",results="show"}
scores_Ca
cat(
    "# scores_kozak_Ca.txt
# Scores of ATG context against Kozak sequence, Candida albicans
# Kozak sequence derived from 400 highest-translated genes by RPF
#
# Gene: Ca systematic gene name
# aATG.scorekn: score of annotated ATG against narrow Kozak sequence (NNNNNATG)
# d1.scorekn: score of 1st downstream ATG against narrow
# xx.scorekw: analogous scores against wide Kozak sequence (NNNNNNNNNATGNNN)
#
# Edward Wallace, Oct 2018
",
file="data_out/scores_kozak_Ca.txt")
scores_Ca %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/scores_kozak_Ca.txt",
              append = TRUE,col_names=TRUE)
```

### Plot against narrow consensus (-5 to ATG)

```{r plot_scores_n_Ca, dependson="calc_scores_Ca", fig.width=5,fig.height=1.8}

sbandwidthn = 0.025

ggplot(data=scores_Ca) +
    geom_density(aes(x=aATG.scorekn,colour="aATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    geom_density(aes(x=d1.scorekn,colour="dATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    # geom_density(aes(x=u1.scorekn,colour="uATG"),
    #             kernel="rectangular",bw=sbandwidthn) +
    geom_density(data=filter(scores_Ca,Gene %in% hiTrans_Ca),
                 aes(x=aATG.scorekn,colour="aATG, top 5%"),
                 kernel="rectangular",bw=sbandwidthn) +
    scale_colour_manual(values=c("aATG"="dodgerblue",
                                 "aATG, top 5%"="darkblue",
                                 # "uATG"="red",
                                 "dATG"="green3") ) +
    labs(x="ATG Kozak score, Sc") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank(),
          legend.title=element_blank())
```

## Compare aATG and dATG context by gene

### Most dATG scores are less than aATG scores

```{r dvsaATG_scoresdens_Ca, dependson="calc_scores_Ca", fig.width=3.5,fig.height=1.8}

ggplot(data=scores_Ca) +
    geom_vline(xintercept=0,size=0.3) +
    geom_density(aes(x=d1vsan,colour="narrow"),kernel="rectangular") +
    # geom_density(aes(x=d1vsaw,colour="wide"),kernel="rectangular")  +
    labs(x="difference in score, dATG vs aATG",
         colour="context") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank(),
          legend.position=c(0.8,0.8))
```

### For highly translated genes, most dATG scores are much less than aATG

```{r dvsaATG_scoresn_Ca, dependson="calc_scores_Ca", fig.width=3.5,fig.height=3.5}
highdiffn_dvsaATG_Ca <- scores_Ca %>%
    arrange(desc(d1vsan)) %>%
    head(n=400) %>%
    select(Gene,aATG.scorekn,d1.scorekn)

ggplot(data=scores_Ca %>% filter(!is.na(d1.scorekn)),
       aes(x=aATG.scorekn,y=d1.scorekn)) +
    geom_diagline() +
    geom_point(size=0.5,colour="grey50") +
    geom_point(data=highdiffn_dvsaATG_Ca,
               colour="red",size=1) +
    geom_point(data=scores_Ca %>% 
                   filter(Gene %in% hiTrans_Ca),
               colour="blue",size=1) +
    geom_point(data=highdiffn_dvsaATG_Ca %>% 
                   filter(Gene %in% hiTrans_Ca),
               colour="purple4",size=1.3) +
    labs(x="aATG score, narrow",
         y="dATG score, narrow")
```

Red: high dATG vs aATG **Kozak** score.
Blue: highly translated.
Purple: both.

### Small negative correlation between dATG and aATG score

R = `r with(scores_Ca, cor(aATG.scorekn,d1.scorekn,use="pairwise.complete.obs")) %>% round3`

## Genes with unusual dATG vs aATG narrow score

Those genes are in this list:

```{r highdiffn_dvsaATG_Ca,dependson="dvsaATG_scoresn_Ca",results="show"}
highdiffn_dvsaATG_Ca
```


Some of these are well translated:

```{r highdiffn_dvsaATG_hiTrans_Ca,dependson="dvsaATG_scoresn_Ca",results="show"}
highdiffn_dvsaATG_Ca %>%
    left_join( ctbl_big_Ca %>% select(Gene, d1.frame,d1.posATG) ) %>% 
    filter(Gene %in% hiTrans_Ca) 
```



## Compare score difference to localization predictions


### Load predictions from mitofates

In input file *CAlbi_mitofates.txt*.

```{r load_mitofates_Ca}
mitofates_Ca <- read_tsv("data_in/CAlbi_mitofates.txt",
                            comment="#",skip=1,
                         col_names=c("Gene","Prob_preseq","Pred_preseq","CSite_enzyme",
"Net_charge","Pos_TOM20","Pos_afhelix",
"BHHPPP","BPHBHH","HBHHBb","HBHHbB","HHBHHB","HHBPHB","HHBPHH",
"HHBPHP","HHHBBH","HHHBPH","HHHHBB","HHPBHH","HPBHHP","PHHBPH")) %>%
    mutate(Pred_preseq=factor(Pred_preseq,
                              levels=c("No mitochondrial presequence",
                                       "Possessing mitochondrial presequence"),
                              labels=c("No","Yes")),
           Gene = str_remove(Gene, " Protein"))
mitocounts_Ca <- 
    mitofates_Ca %>% 
    group_by(Pred_preseq) %>% 
    count() 
mitobreaks_Ca <- mitocounts_Ca %>%
    mutate(label=paste0(Pred_preseq," (n = ",n,")")) %>%
    .$label %>%
    set_names(mitocounts_Ca$Pred_preseq)

mitopredlist_Ca <- mitofates_Ca %>%
    filter(Pred_preseq =="Yes") %>%
    .$Gene
```


### Ca Genes with high dATG vs aATG score might be enriched in mitochondrial presequences?

```{r dvsaATG_scoresdiffn_mitofates_Ca, dependson=c("calc_scores_Ca","load_mitofates_Ca"), fig.width=5,fig.height=1.8}

ggplot(data=scores_Ca %>% 
           filter(!is.na(d1.scorekn)) %>%
           left_join( mitofates_Ca %>% 
                          select(Gene,Prob_preseq,Pred_preseq)
                      ) ,
       aes(x=d1.scorekn-aATG.scorekn,colour=Pred_preseq)) +
    geom_vline(xintercept=0,size=0.3) +
    geom_density(kernel="rectangular",bw=0.025) +
    scale_colour_manual(values=c("Yes"="magenta","No"="grey50"),
                        labels=mitobreaks_Ca)  + 
    labs(x="Kozak score, difference dATG - aATG", colour="Mito. pred") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank())
```

### Ca mito-localized genes also do not have a distinctive aATG context

Although the +5 T is striking.

```{r aATG_motif_mitofates_Ca, dependson=c("calc_scores_Ca","load_mitofates_Ca"), fig.width=4,fig.height=1}

ggseqlogo(data=ctbl_big_Ca %>%
              filter(Gene %in% mitopredlist_Ca) %>%
              .$aATG.context,
          namespace="TCAG")  + 
    theme_nothing()

```

## Genes with unusual dATG vs aATG narrow score AND predicted mito loc

Those genes are in this list:

```{r highdiffn_dvsaATG_mitopred_Ca,dependson="dvsaATG_scoresn_Ca",results="show"}
highdiffn_dvsaATG_Ca %>%
    left_join( ctbl_big_Ca %>% select(Gene, d1.frame,d1.posATG) ) %>%
    left_join( ribosum_Ca ) %>%
    filter(Gene %in% mitopredlist_Ca) %>%
    print(n=26)
```

Only some have in-frame ATG.

* C2_06190W_A / ILV1Putative threonine dehydratase; regulated by *Gcn4* and Gcn2; induced by amino acid starvation (3-AT);


### Some more decently-translated ones with in-frame aATG

```{r highdiffn_dvsaATG_mitopred_hiTrans_Ca,dependson="dvsaATG_scoresn_Ca",results="show"}
highdiffn_dvsaATG_Ca %>%
    left_join( ctbl_big_Ca %>% select(Gene, d1.frame,d1.posATG) ) %>%
    left_join( ribosum_Ca ) %>%
    filter(Gene %in% mitopredlist_Ca, Gene %in% midTrans_Ca, d1.frame==0) 
```

* C2_09260C_A GLO2 hydroxyacylglutathione hydrolase, Scer has GLO2 (cyto) and GLO4 (mito) paralogs
* C1_00490C_A TTR1 Putative glutaredoxin, regulated by Gcn2 and Gcn4, Scer has GRX1 (cyto/nuc) and GRX2 (mito/cyto) paralogs
* C3_02270W_A conserved hypothetical protein with signal peptide, Scer MCO8 is found in mitochondria.
* C6_02410W_A, NAD(+) diphosphatase homolog, Scer NPY1 / YGL067W localizes to peroxisome and cytoplasm.
* C1_08060W_A OFR1 Protein involved in regulation of white-opaque switching
* CR_04300W_A tRNA (cytosine-5-)-methyltransferase, Scer homolog NCL1 is nuclear.
* CR_04590C_A PDX3 Pyridoxamine-phosphate oxidase
* C6_03720W_A ALA1 Alanyl-tRNA synthetase; translational regulation generates cytoplasmic and mitochondrial forms.
* C5_01520C_A GLR1 Glutathione reductase
* C2_10270W_A PUS7 Pseudouridine synthase
* C1_01040W_A Methionine-R-sulfoxide reductase, Scer homolog MXR2 is mitochondrial.
* CR_01550C_A 3'--> 5' exonuclease and endonuclease
* C6_03820C_A conserved protein of unknown function with homology to probable FAD synthase and Molybdenum cofactor biosynth proteins??
* C2_06640C_A tRNA-Val synthetase VAS1
* C6_00390W_A tRNA-Serine synthetase, there is another paralog C3_02780W_A, homology to cyto SES1, and a mito DIA4 homolog.


Several of these are conserved (ALA1, VAS1, PUS7, GLOs)!






<a href="#top">Back to table of contents</a>
