---
title: "Comparisons between Saccharomyces, Transcript Leaders"
author: Edward Wallace, ewjwallace@gmail.com
date: 5 Nov 2019
output:
  # html_notebook: default
  html_document:
    toc: true
    toc_depth: 4
---

# Summary

This summarizes transcript leader/5'UTR lengths in different (annotations of) Saccharomyces species, from yeast genome (YG) and otherwise from Spealman et al 2019 (SM).


* *cer_YG* - S. cerevisiae, Yeast genome (SGD) annotations
* *cer_SM* - S. cerevisiae, Spealman et al annotations
* *kud* - S. kudriavzevii, Spealman et al annotations
* *par* - S. paradoxus, Spealman et al annotations
* *uva* - S. bayanus var uvarum, Spealman et al annotations
    
## Load Packages

```{r setup,message=FALSE,echo=FALSE,cache=FALSE}
knitr::opts_chunk$set(warning=FALSE,message=FALSE,echo=FALSE,cache=TRUE,
                      results="hide",
                      fig.path="figure/CompareUTRlengthSac-",
                      cache.path="cache/CompareUTRlengthSac-")
# load common functions
source("../SharedScripts/sharedScripts.R")
# setwd("~/Repos/CryptoTranscriptome2018/ATGContextOthers/")
```

## Load data

```{r load_data,results="show"}
Orglevels <- c("cer_YG","cer_SM","kud","par","uva")
# infon_Ca_zero <- tibble(Genes="CytoRibo",ATG="aATG",Infon=NA,Organism="Ca")
UTRcolnames <- c("Gene","aATG.pos","Length")

read_ATGcontext_table <- function(file,org) {
  read_tsv(file = file,
           comment = "#",
           skip=1,
           col_names <- c("Gene", "aATG.context", "Length", 
                          "d1.context", "d1.posTSS", "d1.posATG", "d1.frame", 
                          "d2.context", "d2.posTSS", "d2.posATG", "d2.frame", 
                          "u1.context", "u1.posTSS", "u1.posATG", "u1.frame", 
                          "u2.context", "u2.posTSS", "u2.posATG", "u2.frame" ),
           col_types=c("cciciiiciiiciiiciii")) %>%
    dplyr::mutate(Gene=str_remove(Gene,".t0[0-9]")) %>%
    mutate(Organism=org)
}

read_uATGct <- function(file,org) {
  read_tsv(file = file,
           comment = "#") %>%
    mutate(Gene=str_remove(gene,".t0[0-9]")) %>%
    rename_all(str_replace,pattern="-ATG",replacement="") %>%
    select(matches("Gene|(u[0-9+].posTSS)")) %>%
    gather(key="uATG",value="posTSS",-Gene) %>%
    mutate(posTSS=as.integer(posTSS)) %>%
    group_by(Gene) %>%
    summarize(uATGCt = sum(!is.na(posTSS)),
              uATGCtmin20 = sum(posTSS >= 20,na.rm=TRUE)
              ) %>%
    mutate(Organism=org)
}

UTRcts <- bind_rows(
read_uATGct("data_in/SacCer_allATGcontext.table.txt","cer_YG"),
read_uATGct("data_in/SacCerSpeal_allATGcontext.table.txt","cer_SM"),
read_uATGct("data_in/SacKudria_allATGcontext.table.txt","kud"),
read_uATGct("data_in/SacParadoxus_allATGcontext.table.txt","par"),
read_uATGct("data_in/SacBayanus_allATGcontext.table.txt","uva")
) %>%
    mutate(Organism=factor(Organism,levels=Orglevels) )

# read_uATGct("data_in/SacBayanus_allATGcontext.table.txt","bay")

# UTRmess <- readxl::read_excel("data_in/taille des UTRs.xlsx",sheet=2)
UTRlengths <- bind_rows(
read_ATGcontext_table("data_in/SacCer_allATGcontext.table.txt","cer_YG"),
read_ATGcontext_table("data_in/SacCerSpeal_allATGcontext.table.txt","cer_SM"),
read_ATGcontext_table("data_in/SacKudria_allATGcontext.table.txt","kud"),
read_ATGcontext_table("data_in/SacParadoxus_allATGcontext.table.txt","par"),
read_ATGcontext_table("data_in/SacBayanus_allATGcontext.table.txt","uva")
) %>%
    mutate(Organism=factor(Organism,levels=Orglevels) ) %>%
  right_join(UTRcts)

UTRlengths


```

## Only 2-3000 UTRs are annotated in each set

```{r count_UTR_annot,dependson="load_data",results="show"}
UTRsumm <- UTRlengths %>%
  group_by(Organism) %>%
  summarize(uAUGtot=sum(uATGCt,na.rm=TRUE),
            Lengthtot=sum(Length-1,na.rm=TRUE),
            Nlong = sum(Length > 1, na.rm = TRUE), 
            Ntmeas = sum(!is.na(Length)),
            Ntot = length(Length),
            NuAUG = sum(uATGCt>0,na.rm=TRUE))
UTRsumm
```

# Plot UTR length distributions for different species

```{r plot_UTR_lengths,dependson="load_data",fig.height=3.5, fig.width=3.5}

# ggplot(data=UTRlengths,aes(x=Organism,y=Length)) + 
#     geom_violin(aes(colour=Organism)) +
#     scale_color_brewer(type = "qual",palette=1,drop=FALSE) +
#     scale_y_log10nice("TL length",expand=c(0.01,0.01))

plot_UTR_lengths <- 
ggplot(data=UTRlengths %>% filter(Length > 1),
       aes(x=Organism,y=Length-1)) + 
    geom_point(position="jitter",colour="grey50",size=0.015) + 
    geom_boxplot(aes(colour=Organism),fill="NA",outlier.colour = "NA",size=0.8) +
    scale_y_log10("TL length",expand=c(0.01,0.01),
                  breaks=c(1,10,100,1000),
                  labels=c(1,10,100,1000) %>% as.character,
                  oob=scales::squish) +
    scale_color_brewer(type = "qual",palette=1,drop=FALSE) + 
    theme(axis.text.x = element_text(angle=45,vjust=1,hjust=1))
plot_UTR_lengths
# ggplot(data=UTRlengths,aes(x=Organism,y=Length)) + 
#     geom_beeswarm() +
#     scale_y_log10nice()
```

```{r summarize_UTR_lengths,dependson="load_data",results=TRUE}

quantile_tidy <- function(x, probs=c(0.05,0.25,0.5,0.75,0.95)) {
  quantile(x, probs) %>%
    enframe %>%
    list
}

UTRlengths %>%
  filter(Length > 1) %>%
  group_by(Organism) %>%
  summarize(Length=quantile_tidy(Length)) %>%
  unnest %>%
  spread(key=name,value=value) 
```



# Plot uAUG distributions for different species

Proportion of TLs containing uAUGs only includes the TLs with non-zero length, i.e. treating the others as missing data.

```{r plot_uATG_prop,dependson="load_data",fig.height=3.5, fig.width=3.5}

plot_uTL_prop <- 
ggplot(data=UTRsumm,
       aes(x=Organism,y=NuAUG/Nlong,fill=Organism)) + 
    geom_col() +
    scale_y_continuous("Proportion of TLs\ncontaining uAUGs",expand=c(0,0),limits=c(0,1)) +
    scale_fill_brewer(type = "qual",palette=1,drop=FALSE) +
    theme(panel.grid.major.y=element_line(colour="grey50",size=0.2),
          axis.text.x = element_text(angle=45,vjust=1,hjust=1))
plot_uTL_prop

plot_uTL_dens <-
ggplot(data=UTRsumm,aes(x=Organism,y=uAUGtot/Lengthtot,fill=Organism)) +
    geom_col() +
    scale_y_continuous("uAUG density in TLs",expand=c(0,0),limits=c(0,1/64),
                       breaks=c(0,1/128,1/64),labels=c("0","1/128","1/64")) +
    scale_fill_brewer(type = "qual",palette=1,drop=FALSE) +
    theme(panel.grid.major.y=element_line(colour="grey50",size=0.2),
          axis.text.x = element_text(angle=45,vjust=1,hjust=1))

plot_uTL_dens
```

# Supplementary figure: Saccharomyces sensu stricto species have short AUG-poor transcript leaders

We compared transcript leader length (A), proportion of transcript leaders containing uAUGs (B), and density of uAUGs in transcript leaders (C) between annotations of *Saccharomyces* yeasts. Annotations are abbreviated as: cer_YG, *S. cerevisiae* S288C from the saccharomyces genome database (Cherry et al 2013); cer_SM, *S. cerevisiae* S288C from Spealman et al. (2018); kud, *S. kudriavzevii* FM1340, par, *S. paradoxus* CBS432, uva, *S. bayanus *var. *uvarum* JRY9191, the latter 3 also from Spealman et al. Note that all annotations show a similar median leader length of 48-60 nucleotides.


```{r plot_comp,dependson=c("plot_UTR_lengths","plot_uATG_prop"),fig.width=7.5,fig.height=3.5,dev="svg"}
plot_grid(plot_UTR_lengths + theme(legend.position="none"),
          plot_uTL_prop + theme(legend.position="none"),
          plot_uTL_dens + theme(legend.position="none"),
          # get_legend(plot_uTL_dens),
          rel_widths = c(1.2,1,0.99),
          labels=c("A","B","C"),
          align = 'vh',
          nrow=1)
```


