---
title: "Start codon context for Histoplasma capsulatum G217B"
author: Edward Wallace, ewjwallace@gmail.com
date: 16 Jan 2019
output:
  # html_notebook: default
  html_document:
    toc: true
    toc_depth: 4
---

# Summary

This presents the analysis of Histoplasma capsulatum start codon usage and context. This uses the H H. capsulatum G217B translation estimates from Gilmore, Voorhies, Gebhart and Sil 2017.

We decided that the 5'UTR annotations are so patchy that we don't want to pursue this data. Only about half the highly-translated genes have 5'UTRs in the annotation, and there are many contigs and plenty of missing genes including 6 ribosomal proteins.

### Load Packages

```{r setup,message=FALSE,echo=FALSE,cache=FALSE}
knitr::opts_chunk$set(warning=FALSE,message=FALSE,echo=FALSE,cache=TRUE,
                      results="hide",
                      fig.path="figure/HistoATGContext-",
                      cache.path="cache/HistoATGContext-")
# load common functions
source("../SharedScripts/sharedScripts.R")
# setwd("~/Repos/CryptoTranscriptome2018/ATGContextOthers/")
```


# Here we go

## Expression: RNA abundance and ribosome-protected-fragments

### Load expression data

```{r load_ribosum,results="show"}

fpkms <- read_tsv("data_in/ExpressionTranslation/Gilmore2015_GSE68705_Supplementary_File_27.txt") 

names(fpkms) <- str_replace_all(names(fpkms), c("G217B "=""," "="_"))

ribosum <- fpkms %>%
    select(Gene=predicted_gene,
           RNA_Y=Y_mRNA_FPKM,
           RNA_H=H_mRNA_FPKM,
           RPF_Y=Y_footprint_FPKM,
           RPF_H=H_footprint_FPKM) %>%
    mutate(RNA=(RNA_Y+RNA_H)/2,RPF=(RPF_Y+RPF_H)/2,)
    
ribosum %>%
    arrange(desc(RPF)) 

hiTrans <- ribosum %>%
    arrange(desc(RPF)) %>%
    filter(!is.na(Gene)) %>%
    head(n=360) %>%
    .$Gene

hiTrans_Y <- ribosum %>%
    arrange(desc(RPF_Y)) %>%
    head(n=360) %>%
    .$Gene

hiTrans_H <- ribosum %>%
    arrange(desc(RPF_H)) %>%
    head(n=360) %>%
    .$Gene

midTrans <- ribosum %>%
    arrange(desc(RPF)) %>%
    head(n=1437) %>%
    .$Gene

enoughRNA <- ribosum %>%
    arrange(desc(RNA)) %>%
    head(n=3600) %>%
    .$Gene
```


## ATG Context

### Load context data

```{r load_context,results="show"}
ctbl_big <- read_ATGcontext_table("data_in/Histo_allATGcontext.table.txt") %>%
    filter(nchar(aATG.context) == 24) %>%
    mutate(Gene=str_remove(Gene,"_mrna"))
ctbl_big
```

### Annotated ATGs have a Kozak consensus sequence

```{r aATG_context,fig.width=4,fig.height=1}
ggseqlogo(data=ctbl_big$aATG.context,namespace="TCAG") + 
    theme_nothing()
```

### Highly translated Annotated ATGs have a Kozak consensus sequence

That's for hiTrans, the top 5% (256) translated genes by RPF TPM.

```{r a_hi_context,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big,Gene %in% hiTrans)$aATG.context,
                      namespace="TCAG") + 
    theme_nothing()
```

#### Consensus is more visible on genes with 5'UTR annotation.

```{r a_hi_context_split,dependson="load_context",fig.width=4,fig.height=1}
nannotated = ctbl_big %>%
              filter(aATG.pos > 1,Gene %in% hiTrans) %>%
    nrow

nmissing = ctbl_big %>%
              filter(aATG.pos == 1,Gene %in% hiTrans) %>%
    nrow

ggseqlogo(data=ctbl_big %>%
              filter(aATG.pos > 1,Gene %in% hiTrans) %>%
              .$aATG.context,
          namespace="TCAG") +
    theme_nothing() +
    annotate(geom="text",x=22,y=1.2,
             label=paste("5' UTR\nannotated\nn =",nannotated))

ggseqlogo(data=ctbl_big %>%
              filter(aATG.pos==1,Gene %in% hiTrans) %>%
              .$aATG.context,
          namespace="TCAG") +
    theme_nothing() +
    annotate(geom="text",x=22,y=1.2,
             label=paste("5' UTR\nmissing\nn =",nmissing))
```


### Cytoplasmic Ribosome Annotated ATGs have a Kozak consensus sequence

```{r cytoribo_context,dependson="load_context",fig.width=4,fig.height=1}
cytoribolist <- read_tsv("data_in/Histo_CytoRiboList.txt",
                        comment="#") %>%
    filter(!is.na(GENE_ID)) %>%
    .$GENE_ID %>%
    str_remove("_mrna")

ggseqlogo(data=filter(ctbl_big,Gene %in% cytoribolist)$aATG.context,
                      namespace="TCAG") +
    theme_nothing()
```

#### Consensus is more visible on genes with 5'UTR annotation.


```{r cytoribo_context_split,dependson="load_context",fig.width=4,fig.height=1}
nannotated = ctbl_big %>%
              filter(aATG.pos > 1,Gene %in% cytoribolist) %>%
    nrow

nmissing = ctbl_big %>%
              filter(aATG.pos == 1,Gene %in% cytoribolist) %>%
    nrow

ggseqlogo(data=ctbl_big %>%
              filter(aATG.pos > 1,Gene %in% cytoribolist) %>%
              .$aATG.context,
          namespace="TCAG") +
    theme_nothing() +
    annotate(geom="text",x=22,y=1.2,
             label=paste("5' UTR\nannotated\nn =",nannotated))

ggseqlogo(data=ctbl_big %>%
              filter(aATG.pos==1,Gene %in% cytoribolist) %>%
              .$aATG.context,
          namespace="TCAG") +
    theme_nothing() +
    annotate(geom="text",x=22,y=1.2,
             label=paste("5' UTR\nmissing\nn =",nmissing))
```


### Upstream ATGs don't have a consensus

First upstream ATG

```{r u1_context,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big,nchar(u1.context) ==24)$u1.context,
                      namespace="TCAG") + 
    theme_nothing()
```

### Most cytoribo genes are highly translated

Venn diagram 

```{r hitrans_cytoribo_venn,dependson=c("cytoribo_context","load_ribosum"),fig.width=2.1,fig.height=2}
venn(hiTrans=hiTrans,cytoRibo=cytoribolist)
```


### Downstream ATGs don't have a consensus

First downstream ATG

```{r d1_context,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big,nchar(d1.context) ==24)$d1.context,
                      namespace="TCAG") + 
    theme_nothing()
```

### Downstream ATGs in frame and highly translated don't have a consensus

Except for 3rd-codon-position bias.

```{r d1_hi_frame0_context,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big,
                      nchar(d1.context) == 24,
                      d1.frame==0,
                      Gene %in% hiTrans)$d1.context,
                      namespace="TCAG") + 
    theme_nothing()
```

## Calculate Information content and scores of consensus motif

### Calculate a wide and a narrow consensus sequence

Calculate motif score against the position weight matrix (pwm) for both **narrow** (-4 from ATG through to ATG) and **wide** (-10 from ATG) kozak consensus motif. These motifs are taken from the top 5% highly translated genes.

```{r kozak_consensus,fig.width=4,fig.height=1}
kozak_all_PFM <- filter(ctbl_big,Gene %in% hiTrans) %>%
    .$aATG.context %>%
    
    str_to_upper() %>%
    consensusMatrix() 

kozak_w_PWM <- PWM(kozak_all_PFM[,3:15])
kozak_n_PWM <- PWM(kozak_all_PFM[,9:15])
```


```{r write_PFM, dependson="kozak_consensus",results="show"}
cat(
    "# PFM_hiTrans_Hc.txt
# Position Frequency Matrix for aATGs of 256 highest-translated genes from Hc
# This is of the wide ATG context NNNNNNNNNNNNATGNNNNNNNNN
# Column is position, Row is nt count in order: ACGT.
#
# Edward Wallace, June 2018
",
file="data_out/PFM_hiTrans_Hc.txt")
kozak_all_PFM %>%
    as.tibble() %>%
    write_tsv("data_out/PFM_hiTrans_Hc.txt",
              append = TRUE,col_names=FALSE)
```

### Estimate the information content 

Using the sequence logo, details on https://en.wikipedia.org/wiki/Sequence_logo

```{r kozak_inf,dependson=c("kozak_consensus","cytoribo_context"),fig.width=4,fig.height=1,results="show"}
inf_content_narrow <- 
    tribble(
        ~Genes, ~ATG, ~Width, ~Infon,
        "All", "aATG", "narrow", 
        ctbl_big %>%
            .$aATG.context %>%
            infonfromseqs(sstart=9L,send=12L),
        "HiTrans", "aATG", "narrow", 
        ctbl_big %>%
            filter(Gene %in% hiTrans) %>% 
            .$aATG.context %>%
            infonfromseqs(sstart=9L,send=12L),
        "CytoRibo", "aATG", "narrow",
        ctbl_big %>%
            filter(Gene %in% cytoribolist) %>%
            .$aATG.context %>%
            infonfromseqs(sstart=9L,send=12L),
        "All", "d1ATG", "narrow", 
        ctbl_big %>%
            .$d1.context %>%
            na.omit() %>%
            infonfromseqs(sstart=9L,send=12L),
        "HiTrans", "d1ATG", "narrow", 
        ctbl_big %>%
            filter(Gene %in% hiTrans) %>% 
            .$d1.context %>%
            na.omit() %>%
            infonfromseqs(sstart=9L,send=12L),
        "CytoRibo", "d1ATG", "narrow",
        ctbl_big %>%
            filter(Gene %in% cytoribolist) %>%
            .$d1.context %>%
            na.omit() %>%
            infonfromseqs(sstart=9L,send=12L)
    )

write_tsv(inf_content_narrow %>%
              mutate_if(is.numeric,round3),
          "data_out/inf_content_narrow_Hc.txt")
inf_content_narrow
```

Information content in bits of highly-translated consensus (excluding 6 bits from ATG), **narrow** is `r infontot(kozak_all_PFM[,5:9]) %>% round(2)`, of **wide** is `r infontot(kozak_all_PFM[,-(10:12)]) %>% round(2)`.

### Calculate scores of aATG, dATG, uATG against Kozak consensus

```{r calc_scores, dependson="kozak_consensus"}
scores <- 
    ctbl_big %>%
    transmute(Gene,
              aATG.scorekn = PWMscoren(str_to_upper(aATG.context),kozak_n_PWM),
              d1.scorekn   = PWMscoren(d1.context,kozak_n_PWM),
              u1.scorekn   = PWMscoren(str_to_upper(u1.context),kozak_n_PWM),
              aATG.scorekw = PWMscorew(str_to_upper(aATG.context),kozak_w_PWM),
              d1.scorekw   = PWMscorew(d1.context,kozak_w_PWM),
              u1.scorekw   = PWMscorew(str_to_upper(u1.context),kozak_w_PWM),
              NULL
    ) %>%
    mutate(d1vsan = d1.scorekn - aATG.scorekn,
           u1vsan = u1.scorekn - aATG.scorekn,
           d1vsaw = d1.scorekw - aATG.scorekw,
           u1vsaw = u1.scorekw - aATG.scorekw,
           NULL) 

# ribotbl_scoresenough <-
#     ribosum %>%
#     filter(Gene %in% enoughRNA) %>%
#     # left_join(ribotbl,by="Gene") %>% 
#     left_join(scores,by="Gene")

```

We calculate scores using Biostrings::PWMscoreStartingAt.

The best description I could find of this method is: https://support.bioconductor.org/p/61520/

It is just the sum of the matrix product of the PWM with the sequence.


Write scores to file *scores_kozak_Hc.txt*.

```{r write_scores, dependson="calc_scores",results="show"}
scores
cat(
    "# scores_kozak_Hc.txt
# Scores of ATG context against Kozak sequence, H. capsulatum G217B
# Kozak sequence derived from 256 highest-translated genes by RPF
#
# Gene: Hc systematic gene name
# aATG.scorekn: score of annotated ATG against narrow Kozak sequence (NNNNATG)
# d1.scorekn: score of 1st downstream ATG against narrow
# xx.scorekw: analogous scores against wide Kozak sequence (NNNNNNNNNNATG)
#
# Edward Wallace, June 2018
",
file="data_out/scores_kozak_Hc.txt")
scores %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/scores_kozak_Hc.txt",
              append = TRUE,col_names=TRUE)
```

### Plot against narrow consensus (-4 to ATG)

```{r plot_scores_n, dependson="calc_scores", fig.width=5,fig.height=1.8}

sbandwidthn = 0.025

ggplot(data=scores) +
    geom_density(aes(x=aATG.scorekn,colour="aATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    geom_density(aes(x=d1.scorekn,colour="dATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    geom_density(aes(x=u1.scorekn,colour="uATG"),
                kernel="rectangular",bw=sbandwidthn) +
    geom_density(data=filter(scores,Gene %in% hiTrans),
                 aes(x=aATG.scorekn,colour="aATG, top 5%"),
                 kernel="rectangular",bw=sbandwidthn) +
    scale_colour_manual(values=c("aATG"="dodgerblue",
                                 "aATG, top 5%"="darkblue",
                                 "uATG"="red",
                                 "dATG"="green3") ) +
    labs(x="ATG narrow Kozak score, Hc") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank(),
          legend.title=element_blank())
```


### Plot against wide consensus (-10 to ATG)

```{r plot_scores_w, dependson="calc_scores", fig.width=5,fig.height=1.8}

sbandwidthn = 0.025

ggplot(data=scores) +
    geom_density(aes(x=aATG.scorekw,colour="aATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    geom_density(aes(x=d1.scorekw,colour="dATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    geom_density(aes(x=u1.scorekw,colour="uATG"),
                kernel="rectangular",bw=sbandwidthn) +
    geom_density(data=filter(scores,Gene %in% hiTrans),
                 aes(x=aATG.scorekw,colour="aATG, top 5%"),
                 kernel="rectangular",bw=sbandwidthn) +
    scale_colour_manual(values=c("aATG"="dodgerblue",
                                 "aATG, top 5%"="darkblue",
                                 "uATG"="red",
                                 "dATG"="green3") ) +
    labs(x="ATG wide Kozak score, Hc") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank(),
          legend.title=element_blank())
```


## Compare aATG and dATG context by gene

### Most dATG scores are less than aATG scores

```{r dvsaATG_scoresdens, dependson="calc_scores", fig.width=3.5,fig.height=1.8}

ggplot(data=scores) +
    geom_vline(xintercept=0,size=0.3) +
    geom_density(aes(x=d1vsan,colour="narrow"),kernel="rectangular") +
    # geom_density(aes(x=d1vsaw,colour="wide"),kernel="rectangular")  +
    labs(x="difference in score, dATG vs aATG",
         colour="context") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank(),
          legend.position=c(0.8,0.8))
```

### For highly translated genes, most dATG narrow scores are much less than aATG

```{r dvsaATG_scoresn, dependson="calc_scores", fig.width=3.5,fig.height=3.5}
highdiffn_dvsaATG <- scores %>%
    arrange(desc(d1vsan)) %>%
    head(n=400) %>%
    select(Gene,aATG.scorekn,d1.scorekn)

ggplot(data=scores %>% filter(!is.na(d1.scorekn)),
       aes(x=aATG.scorekn,y=d1.scorekn)) +
    geom_diagline() +
    geom_point(size=0.5,colour="grey50") +
    geom_point(data=highdiffn_dvsaATG,
               colour="red",size=1) +
    geom_point(data=scores %>% 
                   filter(Gene %in% hiTrans),
               colour="blue",size=1) +
    geom_point(data=highdiffn_dvsaATG %>% 
                   filter(Gene %in% hiTrans),
               colour="purple4",size=1.3) +
    labs(x="aATG score, narrow",
         y="dATG score, narrow")
```

Red: high dATG vs aATG **Kozak** score.
Blue: highly translated.
Purple: both.

### Small negative correlation between dATG and aATG score narrow

R = `r with(scores, cor(aATG.scorekn,d1.scorekn,use="pairwise.complete.obs")) %>% round3`

### For highly translated genes, most dATG wide scores are much less than aATG

```{r dvsaATG_scoresw, dependson="calc_scores",fig.width=3.5,fig.height=3.5}
highdiffw_dvsaATG <- scores %>%
    arrange(desc(d1vsaw)) %>%
    head(n=400) %>%
    select(Gene,aATG.scorekw,d1.scorekw)

ggplot(data=scores %>% filter(!is.na(d1.scorekw)),
       aes(x=aATG.scorekw,y=d1.scorekw)) +
    geom_diagline() +
    geom_point(size=0.5,colour="grey50") +
    geom_point(data=highdiffw_dvsaATG,
               colour="red",size=1) +
    geom_point(data=scores %>%
                   filter(Gene %in% hiTrans),
               colour="blue",size=1) +
    geom_point(data=highdiffw_dvsaATG %>%
                   filter(Gene %in% hiTrans),
               colour="purple4",size=1.3) +
    labs(x="aATG score, wide",
         y="dATG score, wide")
```

Red: high dATG vs aATG **Kozak** score.
Blue: highly translated.
Purple: both.

### Small negative correlation between dATG and aATG score wide

R = `r with(scores, cor(aATG.scorekw,d1.scorekw,use="pairwise.complete.obs")) %>% round3`

## Genes with unusual dATG vs aATG narrow score

Those genes are in this list:

```{r highdiffn_dvsaATG,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG
```

Some of these are highly translated (top 5%)

```{r highdiffn_dvsaATG_hiTrans,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG %>% 
                   filter(Gene %in% hiTrans) 
```


### dATG in frame with ATG, narrow

Files with high difference in narrow score, filtered for reasonable amounts of RNA, in frame. Saved to *dvsaATG_highdiffn_inframe_Hc.txt*.

```{r highdiffn_dvsaATG_inframe,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG_enoughinframe <- 
    highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene,d1.frame) ) %>%
    filter(Gene %in% enoughRNA, d1.frame==0) %>%
    select(Gene,aATG.scorekn,d1.scorekn)

highdiffn_dvsaATG_enoughinframe

cat(
    "# dvsaATG_highdiffn_inframe_Hc.txt
# Genes with in-frame high-score dATGs
# In H. capsulatum G217B:
#   - in top 5% by difference, aATG.scorekn - d1.scorekn
#   - in top 50% by RNA abundance
#   - dATG is in frame with aATG
#
# Gene: systematic gene name
# aATG.scorekn: annotated ATG score against narrow Kozak consensus sequence
# d1.scorekn: 1st downstream ATG score against narrow Kozak consensus sequence
#
# Edward Wallace, Dec 2018
",
file="data_out/dvsaATG_highdiffn_inframe_Hc.txt")
highdiffn_dvsaATG_enoughinframe %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/dvsaATG_highdiffn_inframe_Hc.txt",append = TRUE,col_names=TRUE)
```


### dATG out of frame with ATG, narrow

Files with high difference in narrow score, filtered for reasonable amounts of RNA, out of frame. Saved to *dvsaATG_highdiffn_outframe_Hc.txt*.


```{r highdiffn_dvsaATG_outframe,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG_enoughoutframe <- 
    highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene,d1.frame) ) %>%
    filter(Gene %in% enoughRNA, d1.frame!=0) %>%
    select(Gene,aATG.scorekn,d1.scorekn)

highdiffn_dvsaATG_enoughoutframe

cat(
    "# dvsaATG_highdiffn_outframe_Hc.txt
# Genes with in-frame high-score dATGs
# In H. capsulatum G217B:
#   - in top 5% by difference, aATG.scorekn - d1.scorekn
#   - in top 50% by RNA abundance
#   - dATG is out-of-frame with aATG
#
# Gene: systematic gene name
# aATG.scorekn: annotated ATG score against narrow Kozak consensus sequence
# d1.scorekn: 1st downstream ATG score against narrow Kozak consensus sequence
#
# Edward Wallace, Dec 2018
",
file="data_out/dvsaATG_highdiffn_outframe_Hc.txt")
highdiffn_dvsaATG_enoughoutframe %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/dvsaATG_highdiffn_outframe_Hc.txt",append = TRUE,col_names=TRUE)
```


### dATG in frame with ATG, wide

Files with high difference in wide score, filtered for reasonable amounts of RNA, in frame. Saved to *dvsaATG_highdiffw_inframe_Hc.txt*.

```{r highdiffw_dvsaATG_inframe,dependson="dvsaATG_scoresn",results="show"}
highdiffw_dvsaATG_enoughinframe <- 
    highdiffw_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene,d1.frame) ) %>%
    filter(Gene %in% enoughRNA, d1.frame==0) %>%
    select(Gene,aATG.scorekw,d1.scorekw)

highdiffw_dvsaATG_enoughinframe

cat(
    "# dvsaATG_highdiffw_inframe_Hc.txt
# Genes with in-frame high-score dATGs
# In H. capsulatum G217B:
#   - in top 5% by difference, aATG.scorekw - d1.scorekw
#   - in top 50% by RNA abundance
#   - dATG is in frame with aATG
#
# Gene: systematic gene name
# aATG.scorekw: annotated ATG score against wide Kozak consensus sequence
# d1.scorekw: 1st downstream ATG score against wide Kozak consensus sequence
#
# Edward Wallace, Jan 2019
",
file="data_out/dvsaATG_highdiffw_inframe_Hc.txt")
highdiffw_dvsaATG_enoughinframe %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/dvsaATG_highdiffw_inframe_Hc.txt",append = TRUE,col_names=TRUE)
```


### dATG out of frame with ATG, wide

Files with high difference in wide score, filtered for reasonable amounts of RNA, out of frame. Saved to *dvsaATG_highdiffw_outframe_Hc.txt*.


```{r highdiffw_dvsaATG_outframe,dependson="dvsaATG_scoresn",results="show"}
highdiffw_dvsaATG_enoughoutframe <- 
    highdiffw_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene,d1.frame) ) %>%
    filter(Gene %in% enoughRNA, d1.frame!=0) %>%
    select(Gene,aATG.scorekw,d1.scorekw)

highdiffw_dvsaATG_enoughoutframe

cat(
    "# dvsaATG_highdiffw_outframe_Hc.txt
# Genes with in-frame high-score dATGs
# In H. capsulatum G217B:
#   - in top 5% by difference, aATG.scorekw - d1.scorekw
#   - in top 50% by RNA abundance
#   - dATG is out-of-frame with aATG
#
# Gene: systematic gene name
# aATG.scorekw: annotated ATG score against wide Kozak consensus sequence
# d1.scorekw: 1st downstream ATG score against wide Kozak consensus sequence
#
# Edward Wallace, Jan 2019
",
file="data_out/dvsaATG_highdiffw_outframe_Hc.txt")
highdiffw_dvsaATG_enoughoutframe %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/dvsaATG_highdiffw_outframe_Hc.txt",append = TRUE,col_names=TRUE)
```


## Compare score difference to localization predictions


### Load predictions from mitofates

In input file *Histo_mitofates.txt*.

```{r load_mitofates}
mitofates <- read_mitofates("data_in/Histo_mitofates.txt") %>%
    mutate(Gene=str_remove(Gene,"_mrna"))

mitocounts <- 
    mitofates %>% 
    group_by(Pred_preseq) %>% 
    count() 
mitobreaks <- mitocounts %>%
    mutate(label=paste0(Pred_preseq," (n = ",n,")")) %>%
    .$label %>%
    set_names(mitocounts$Pred_preseq)

mitopredlist <- mitofates %>%
    filter(Pred_preseq =="Yes") %>%
    .$Gene
```


### Hc Genes with high dATG vs aATG score *are* enriched in mitochondrial presequences

```{r dvsaATG_scoresdiffn_mitofates, dependson=c("calc_scores","load_mitofates"), fig.width=5,fig.height=1.8}

ggplot(data=scores %>% 
           filter(!is.na(d1.scorekn)) %>%
           left_join( mitofates %>% 
                          select(Gene,Prob_preseq,Pred_preseq)
                      ) ,
       aes(x=d1.scorekn-aATG.scorekn,colour=Pred_preseq)) +
    geom_vline(xintercept=0,size=0.3) +
    geom_density(kernel="rectangular",bw=0.025) +
    scale_colour_manual(values=c("Yes"="magenta","No"="grey50"),
                        labels=mitobreaks)  + 
    labs(x="Kozak score, difference dATG - aATG", colour="Mito. pred") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank())
```

### Hc mito-localized genes also do not have a distinctive aATG context

```{r aATG_motif_mitofates, dependson=c("calc_scores","load_mitofates"), fig.width=4,fig.height=1}

ggseqlogo(data=ctbl_big %>%
              filter(Gene %in% mitopredlist) %>%
              .$aATG.context,
          namespace="TCAG")  + 
    theme_nothing()

```

## Genes with unusual dATG vs aATG narrow score AND predicted mito loc

Those genes are in this list:

```{r highdiffn_dvsaATG_mitopred,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene, d1.frame,d1.posATG) ) %>%
    left_join( ribosum %>% select(Gene,RPF)) %>%
    filter(Gene %in% mitopredlist) 
```

### less strong biases, but in frame dATG

```{r highdiffn_dvsaATG_mitopred_inframe,dependson="dvsaATG_scoresn",results="show"}
scores %>%
    arrange(desc(d1vsan)) %>%
    select(Gene,aATG.scorekn,d1.scorekn) %>%
    inner_join( ctbl_big %>% 
                   filter(d1.frame==0) %>% 
                   select(Gene,d1.posATG),
               by="Gene" ) %>%
    left_join( ribosum  %>% select(Gene,RPF) ) %>%
    filter(Gene %in% mitopredlist) %>%
    print(n=20)
```

<a href="#top">Back to table of contents</a>
