---
title: "Start codon context for Neurospora crassa"
author: Edward Wallace, ewjwallace@gmail.com
date: 16 Jan 2019
output:
  # html_notebook: default
  html_document:
    toc: true
    toc_depth: 4
---

# Summary

This presents the analysis of Neurospora start codon usage and context. This uses the N. crassa translation estimates from Ivanov 2017, strain FGSC 2489. RNA data and TE estimates are from Yu 2015, same strain.

For "enough RNA" we pick the top ~40% (4000) by RNA abundance, because the N. crassa genome is much larger (9722 genes) than the other fungi we are addressing.

### Load Packages

```{r setup,message=FALSE,echo=FALSE,cache=FALSE}
knitr::opts_chunk$set(warning=FALSE,message=FALSE,echo=FALSE,cache=TRUE,
                      results="hide",
                      fig.path="figure/NcrassaATGContext-",
                      cache.path="cache/NcrassaATGContext-")
# load common functions
source("../SharedScripts/sharedScripts.R")
# setwd("~/Repos/CryptoTranscriptome2018/ATGContextOthers/")
```


# Here we go

## Expression: RNA abundance and ribosome-protected-fragments

### Load expression data

```{r load_ribosum,results="show"}

tpms_Yu <- read_tsv("data_in/ExpressionTranslation/Neurospora_Yu2015_TPMs_collated.tsv") %>%
    dplyr::rename(Gene=ORF)

tpms_Iv <- read_tsv("data_in/ExpressionTranslation/Neurospora_Ivanov2017_TPMs_collated.tsv") %>%
    dplyr::rename(Gene=ORF)

ribosum_Yu <- tpms_Yu %>%
    mutate(RPF=(WT+Opt)/2) %>%
    dplyr::rename(RNA=WTRNA)

ribosum_Iv <- tpms_Iv %>%
    mutate(RPF=(WT1A+WT1B+WT2A+WT2B+HS1A+HS1B)/6)

ribosum <- ribosum_Iv

ribosum %>%
    arrange(desc(RPF)) 

hiTrans <- ribosum %>%
    arrange(desc(RPF)) %>%
    head(n=400) %>%
    .$Gene

midTrans <- ribosum %>%
    arrange(desc(RPF)) %>%
    head(n=2000) %>%
    .$Gene

# nonterribleTrans <- ribosum %>%
#     arrange(desc(RPF)) %>%
#     head(n=8000) %>%
#     .$Gene

enoughRNA <- ribosum_Yu %>%
    arrange(desc(RNA)) %>%
    head(n=4000) %>%
    .$Gene
```

Reasonably, NCU01528 is GAPDH. NCU10042 is enolase-1. NCU02003 is elongation factor 1-alpha. NCU02193 is Pdc1. NCU01754 is Adh1.

Weirdly high expression of thiazole/thiamine biosynthetic enzymes NCU06110/thi4, NCU09345/nmt1.

NCU05495/ccg-16 is CVNH domain-containing protein. NCU08332/hex-1 is isworonin body major protein.

Several hypothetical proteins (NCU16635/NCU04276). Wow. check footprint locations.

#### Check RPFs vs RNA in Yu 2015

```{r scatter_ribo_Yu,dependson="load_ribosum",results="show",fig.width=3.9,fig.height=4.1}
# ggplot(data=tpms_Yu) + 
#     geom_point(aes(x=WT,y=Opt)) +
#     scale_loglog()

ggplot(data=tpms_Yu,aes(x=WTRNA,y=WT)) + 
    geom_diagline() + 
    geom_point(colour="grey50",size=0.8) +
    geom_text_repel(aes(label=str_sub(Gene,start=4L)),
                    data=tpms_Yu %>% 
                        filter( (WTRNA > 1e3 & WT < WTRNA / 100) | WT > 1e4 ) 
                    ) + 
    scale_loglog() +
    labs(x="WT RNA TPM", y="WT RPF TPM")
```

Many genes have RNA << RPF.  They could be misannotated, or translationally repressed. E.G. NCU04987, NCU16627, NCU06322, NCU08491.

The numbers for NCU04987 are so absurd that it is surely a misannotation of a ncRNA.  NCU16627 also looks very strange.

```{r cache=FALSE}
tpms_Yu %>%
    filter(Gene %in% c("NCU06110","NCU01528","NCU09345","NCU02003",
                       "NCU01187","NCU04987","NCU16627","NCU06322","NCU08491"))
```


#### Check replicability

```{r scatter_ribo,dependson="load_ribosum",results="show",fig.width=10,fig.height=10}
library(GGally)
ggpairs(left_join(tpms_Yu,tpms_Iv) %>%
            select(-Gene) %>%
            log10()
        )
```

### Ribosome occupancy mostly tracks RNA abundance

```{r plot_TE,dependson="load_ribosum",fig.height=3.9,fig.width=4.1,eval=FALSE}
TE_unusual <- 
    ribosum %>%
    filter( (RPF > 1.8e4) |
                (RPF > 100 & RPF > RNA * 5 ) | 
                (RNA > 150 & RPF < RNA / 10 ))


ggplot(data=ribosum,aes(x=RNA,y=RPF)) +
    geom_diagline() +
    geom_point(size=.8,colour="grey50") +
    geom_point(data=TE_unusual,aes(colour=Gene),size=1) + 
    geom_text_repel(data=TE_unusual,aes(label=Gene,colour=Gene)) +
    scale_loglog(limits=c(1e-1,2e4),
                  oob=scales::squish,
                  expand=c(0.01,0.01)) +
    labs(x="RNA abundance (TPM of aORF)",
         y="Ribosome occupancy (TPM of aORF)") +
    theme(legend.position="none")
```

## ATG Context

### Load context data

```{r load_context,results="show"}
ctbl_big <- read_ATGcontext_table("data_in/NC12_allATGcontext.table.txt")
ctbl_big
```

### Annotated ATGs have a Kozak consensus sequence

```{r aATG_context,fig.width=4,fig.height=1}
ggseqlogo(data=ctbl_big$aATG.context,namespace="TCAG") + 
    theme_nothing()
```

### Highly translated Annotated ATGs have a Kozak consensus sequence

That's for hiTrans, the top 4% (400) translated genes by RPF TPM.

```{r a_hi_context,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big,Gene %in% hiTrans)$aATG.context,
                      namespace="TCAG") + 
    theme_nothing()
```

### Cytoplasmic Ribosome Annotated ATGs have a Kozak consensus sequence

```{r cytoribo_context,dependson="load_context",fig.width=4,fig.height=1}
cytoribolist <- read_tsv("data_out/ComplexOrthologs/cytoribosome_PTRcurated6.txt",
                        comment="#") %>%
    filter(SPECIES=="NEUCR") %>%
    .$GENE_ID
ggseqlogo(data=filter(ctbl_big,Gene %in% cytoribolist)$aATG.context,
                      namespace="TCAG") + 
    theme_nothing()
```

### Almost all cytoribo genes are highly translated

Venn diagram 

```{r hitrans_cytoribo_venn,dependson=c("cytoribo_context","load_ribosum"),fig.width=2.1,fig.height=2}
venn(hiTrans=hiTrans,cytoRibo=cytoribolist)
```

### Downstream ATGs don't have a consensus

First downstream ATG

```{r d1_context,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big,nchar(d1.context) ==24)$d1.context,
                      namespace="TCAG") + 
    theme_nothing()
```

### Downstream ATGs in frame and highly translated don't have a consensus

Except for 3rd-codon-position bias.

```{r d1_hi_frame0_context,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big,
                      nchar(d1.context) == 24,
                      d1.frame==0,
                      Gene %in% hiTrans)$d1.context,
                      namespace="TCAG") + 
    theme_nothing()
```

## Calculate Information content and scores of consensus motif

### Calculate a wide and a narrow consensus sequence

Calculate motif score against the position weight matrix (pwm) for both **narrow** (-4 from ATG through to ATG) and **wide** (-10 from ATG) kozak consensus motif. These motifs are taken from the top 5% highly translated genes.

```{r kozak_consensus,fig.width=4,fig.height=1}
kozak_all_PFM <- filter(ctbl_big,Gene %in% hiTrans) %>%
    .$aATG.context %>%
    
    str_to_upper() %>%
    consensusMatrix() 

kozak_w_PWM <- PWM(kozak_all_PFM[,3:15])
kozak_n_PWM <- PWM(kozak_all_PFM[,9:15])
```


```{r write_PFM, dependson="kozak_consensus",results="show"}
cat(
    "# PFM_hiTrans_Nc.txt
# Position Frequency Matrix for aATGs of 400 highest-translated genes from Nc
# This is of the wide ATG context NNNNNNNNNNNNATGNNNNNNNNN
# Column is position, Row is nt count in order: ACGT.
#
# Edward Wallace, June 2018
",
file="data_out/PFM_hiTrans_Nc.txt")
kozak_all_PFM %>%
    as.tibble() %>%
    write_tsv("data_out/PFM_hiTrans_Nc.txt",
              append = TRUE,col_names=FALSE)
```

### Estimate the information content 

Using the sequence logo, details on https://en.wikipedia.org/wiki/Sequence_logo

```{r kozak_inf,dependson=c("kozak_consensus","cytoribo_context"),fig.width=4,fig.height=1,results="show"}
inf_content_narrow <- 
    tribble(
        ~Genes, ~ATG, ~Width, ~Infon,
        "All", "aATG", "narrow", 
        ctbl_big %>%
            .$aATG.context %>%
            infonfromseqs(sstart=9L,send=12L),
        "HiTrans", "aATG", "narrow", 
        ctbl_big %>%
            filter(Gene %in% hiTrans) %>% 
            .$aATG.context %>%
            infonfromseqs(sstart=9L,send=12L),
        "CytoRibo", "aATG", "narrow", 
        ctbl_big %>%
            filter(Gene %in% cytoribolist) %>% 
            .$aATG.context %>%
            infonfromseqs(sstart=9L,send=12L),
        "All", "d1ATG", "narrow", 
        ctbl_big %>%
            .$d1.context %>%
            na.omit() %>%
            infonfromseqs(sstart=9L,send=12L),
        "HiTrans", "d1ATG", "narrow", 
        ctbl_big %>%
            filter(Gene %in% hiTrans) %>% 
            .$d1.context %>%
            na.omit() %>%
            infonfromseqs(sstart=9L,send=12L),
        "CytoRibo", "d1ATG", "narrow", 
        ctbl_big %>%
            filter(Gene %in% cytoribolist) %>% 
            .$d1.context %>%
            na.omit() %>%
            infonfromseqs(sstart=9L,send=12L)
    )

write_tsv(inf_content_narrow %>%
              mutate_if(is.numeric,round3),
          "data_out/inf_content_narrow_Nc.txt")
inf_content_narrow
```

Information content in bits of highly-translated consensus (excluding 6 bits from ATG), **narrow** is `r infontot(kozak_all_PFM[,5:9]) %>% round(2)`, of **wide** is `r infontot(kozak_all_PFM[,-(10:12)]) %>% round(2)`.


### Estimate information content per base across start context

```{r kozak_inf_perbase,dependson=c("kozak_consensus","cytoribo_context"),fig.width=6,fig.height=1.4,results="show"}
inf_content_perbase <- 
    tibble(i = 1:24, 
           Pos = c(-12:-1, 1:12),
        All = ctbl_big %>%
            .$aATG.context %>%
            infonfromseqs(scope="all"),
        HiTrans = ctbl_big %>%
            filter(Gene %in% hiTrans) %>% 
            .$aATG.context %>%
            infonfromseqs(scope="all"),
        CytoRibo = ctbl_big %>%
            filter(Gene %in% cytoribolist) %>% 
            .$aATG.context %>%
            infonfromseqs(scope="all")
    )

write_tsv(inf_content_perbase %>%
              mutate_if(is.numeric,round3),
          "data_out/inf_content_perbase_Nc.txt")
# inf_content_perbase

plot_inf_perbase(inf_content_perbase)
```

### Calculate scores of aATG, dATG, uATG against Kozak consensus

```{r calc_scores, dependson="kozak_consensus"}
scores <- 
    ctbl_big %>%
    transmute(Gene,
              aATG.scorekn = PWMscoren(str_to_upper(aATG.context),kozak_n_PWM),
              d1.scorekn   = PWMscoren(d1.context,kozak_n_PWM),
              u1.scorekn   = PWMscoren(u1.context,kozak_n_PWM),
              aATG.scorekw = PWMscorew(str_to_upper(aATG.context),kozak_w_PWM),
              d1.scorekw   = PWMscorew(d1.context,kozak_w_PWM),
              u1.scorekw   = PWMscorew(u1.context,kozak_w_PWM),
              NULL
    ) %>%
    mutate(d1vsan = d1.scorekn - aATG.scorekn,
           u1vsan = u1.scorekn - aATG.scorekn,
           d1vsaw = d1.scorekw - aATG.scorekw,
           u1vsaw = u1.scorekw - aATG.scorekw,
           NULL) 


# ribotbl_scoresenough <-
#     ribosum %>%
#     filter(Gene %in% enoughRNA) %>%
#     # left_join(ribotbl,by="Gene") %>% 
#     left_join(scores,by="Gene")

```

We calculate scores using Biostrings::PWMscoreStartingAt.

The best description I could find of this method is: https://support.bioconductor.org/p/61520/

It is just the sum of the matrix product of the PWM with the sequence.


Write scores to file *scores_kozak_Nc.txt*.

```{r write_scores, dependson="calc_scores",results="show"}
scores
cat(
    "# scores_kozak_Nc.txt
# Scores of ATG context against Kozak sequence, N. crassa 
# Kozak sequence derived from 400 highest-translated genes by RPF
#
# Gene: Nc systematic gene name
# aATG.scorekn: score of annotated ATG against narrow Kozak sequence (NNNNATG)
# d1.scorekn: score of 1st downstream ATG against narrow
# xx.scorekw: analogous scores against wide Kozak sequence (NNNNNNNNNNATG)
#
# Edward Wallace, June 2018
",
file="data_out/scores_kozak_Nc.txt")
scores %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/scores_kozak_Nc.txt",
              append = TRUE,col_names=TRUE)
```

### Plot against narrow consensus (-4 to ATG)

```{r plot_scores_n, dependson="calc_scores", fig.width=5,fig.height=1.8}

sbandwidthn = 0.025

ggplot(data=scores) +
    geom_density(aes(x=aATG.scorekn,colour="aATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    geom_density(aes(x=d1.scorekn,colour="dATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    # geom_density(aes(x=u1.scorekn,colour="uATG"),
    #             kernel="rectangular",bw=sbandwidthn) +
    geom_density(data=filter(scores,Gene %in% hiTrans),
                 aes(x=aATG.scorekn,colour="aATG, top 5%"),
                 kernel="rectangular",bw=sbandwidthn) +
    scale_colour_manual(values=c("aATG"="dodgerblue",
                                 "aATG, top 5%"="darkblue",
                                 # "uATG"="red",
                                 "dATG"="green3") ) +
    labs(x="ATG narrow Kozak score, Nc") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank(),
          legend.title=element_blank())
```


### Plot against wide consensus (-10 to ATG)

```{r plot_scores_w, dependson="calc_scores", fig.width=5,fig.height=1.8}

sbandwidthn = 0.025

ggplot(data=scores) +
    geom_density(aes(x=aATG.scorekw,colour="aATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    geom_density(aes(x=d1.scorekw,colour="dATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    # geom_density(aes(x=u1.scorekw,colour="uATG"),
    #             kernel="rectangular",bw=sbandwidthn) +
    geom_density(data=filter(scores,Gene %in% hiTrans),
                 aes(x=aATG.scorekw,colour="aATG, top 5%"),
                 kernel="rectangular",bw=sbandwidthn) +
    scale_colour_manual(values=c("aATG"="dodgerblue",
                                 "aATG, top 5%"="darkblue",
                                 "uATG"="red",
                                 "dATG"="green3") ) +
    labs(x="ATG wid Kozak score, Nc") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank(),
          legend.title=element_blank())
```


## Mutual information of different positions around ATG

### For all annotated genes

```{r mutinf,dependson=c("load_context","load_riboviz"),results="show",fig.height=4,fig.width=5}
mutinf <- mutinfostr(ctbl_big$aATG.context)
# mutinf

plot_mutinfo(mutinf)
```

### For highly translated genes

```{r mutinf_hiTrans,dependson=c("load_context","load_riboviz"),results="show",fig.height=4,fig.width=5}
ctbl_big %>%
    filter(Gene %in% hiTrans) %>%
    .$aATG.context %>%
    mutinfostr %>%
    plot_mutinfo
```


### For cytoplasmic ribosomal proteins, strong correlation between adjacent bases

This suggests the idea that an upstream CCG motif (see the logo) compensates for A's in the classic Kozak positions closer to the ATG.

```{r mutinf_cytoribo,dependson=c("load_context","load_riboviz"),results="show",fig.height=4,fig.width=5}
ctbl_big %>%
    filter(Gene %in% cytoribolist) %>%
    .$aATG.context %>%
    mutinfostr %>%
    plot_mutinfo

```


### What is the -2 correlation in cyto ribo genes?

```{r cytoribo_contextm2,dependson="load_context",fig.width=4,fig.height=3}

seqlogoFilterGenePos <- function(datain=ctbl_big,Genelist=hiTrans,
                                 pos=8L, nt="T") {
    datafiltered <- filter(datain,
                          Gene %in% Genelist,
                          str_sub(aATG.context,start=pos,end=pos)==nt
    )
    ggseqlogo(data=datafiltered$aATG.context,
    namespace="TCAG") + 
        theme_nothing() +
        annotate(geom="text",x=21,y=1,
                 label=paste0("n = ", nrow(datafiltered)))
}

plot_grid(seqlogoFilterGenePos(Genelist=cytoribolist,pos=11L,nt="T"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=11L,nt="C"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=11L,nt="A"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=11L,nt="G"),
    ncol=1
)
    
```

There are definitely correlations there.


### What is the -4 correlation in cyto ribo genes?

```{r cytoribo_contextm4,dependson="load_context",fig.width=4,fig.height=3}

plot_grid(seqlogoFilterGenePos(Genelist=cytoribolist,pos=9L,nt="T"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=9L,nt="C"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=9L,nt="A"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=9L,nt="G"),
    ncol=1
)
    
```


### What is the -8 correlation in cyto ribo genes?

```{r cytoribo_contextm8,dependson="load_context",fig.width=4,fig.height=3}

plot_grid(seqlogoFilterGenePos(Genelist=cytoribolist,pos=5L,nt="T"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=5L,nt="C"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=5L,nt="A"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=5L,nt="G"),
    ncol=1
)
    
```


## uATGs inhibit translation of the main ORF

### uATGs are associated with slightly lower absolute translation

This is a very weak effect.

```{r uATG_RPF, dependson="calc_scores", fig.width=2.5,fig.height=3.5}
uATG_ct <- read_tsv("data_in/NC12_allATGcontext.table.txt") %>% 
    dplyr::rename(Gene=gene) %>%
    dplyr::rename_all(str_replace,pattern="-ATG",replacement="") %>%
    dplyr::select(c("Gene",paste0("u",1:56,".posTSS"))) %>%
    gather(key="uATG",value="posTSS",-Gene) %>%
    mutate(posTSS=as.integer(posTSS)) %>%
    group_by(Gene) %>%
    summarize(uATGCt = sum(!is.na(posTSS)),
              uATGCtmin20 = sum(posTSS >= 20,na.rm=TRUE)
              ) %>%
    mutate(uATGCtC = replace(uATGCt,uATGCt>=2, "2+") %>%
               factor(levels=c(0:1,"2+")),
           uATGCtmin20C = replace(uATGCtmin20,uATGCtmin20>=2, "2+") %>%
               factor(levels=c(0:1,"2+")),
           uATGfac = make_uATGfact(uATGCt,uATGCtmin20)
           )

ribo_uct <- 
    ribosum %>%
    left_join( ribosum_Yu %>% 
                   mutate(TE = RPF / RNA) %>%
                   select(Gene,TE) ) %>%
    filter(Gene %in% enoughRNA) %>%
    inner_join(uATG_ct)

ggplot(data=ribo_uct,
       aes(x=uATGCtC,y=RPF)) +
    geom_jitter(colour="grey50",size=0.7) +
    geom_boxplot(colour="blue",outlier.colour=NA,fill=NA,size=0.75) +
    scale_y_log10nice("Ribosome occupancy (TPM of aORF)",
                      limits=c(1,10000),
                      oob=scales::squish) +
    labs(x="uATG count")
```

### uATGs are not associated with lower translation efficiency

But TE is from single-replicate data Yu 2015 so it may be difficult to draw conclusions.

```{r uATG_TE, dependson="uATG_RPF", fig.width=2.5,fig.height=3.5}

ggplot(data=ribo_uct,
       aes(x=uATGCtC,y=TE)) +
    geom_jitter(colour="grey50",size=0.7) +
    geom_boxplot(colour="blue",outlier.colour=NA,fill=NA,size=0.75) +
    scale_y_log10nice("TE of aORF",
                      limits=c(0.05,10),expand=c(0,0),
                      oob=scales::squish) +
    labs(x="uATG count")
```

### uATGs association with translation efficiency does not vary with 20nt from TSS

```{r uATGmin20_TE, dependson="uATG_RPF", fig.width=2.5,fig.height=3.5,results="show"}
# ribo_uct
ggplot(data=ribo_uct,
       aes(x=uATGfac,y=TE)) +
    geom_jitter(colour="grey50",size=0.7) +
    geom_boxplot(colour="blue",outlier.colour=NA,fill=NA,size=0.75) +
    scale_y_log10nice("TE of aORF",
                      limits=c(0.05,10),expand=c(0,0),
                      oob=scales::squish) +
    labs(x="uATG count and position\nrelative to TSS")
```

### uATG score weakly affects TE

We suspect that uATG is associated with lower TE if the uATG has

* position at least 20nt downstream from TSS
* higher score

This figure shows that, *for genes with only 1 uATG*, this correlation is weak but as expected. We would expect a stronger effect for genes with multiple uATGs of higher score. If we could model more aspects of translational efficiency we might find a stronger link.

```{r uATG_scoreTE, dependson="uATG_RPF", fig.width=4.5,fig.height=2.5,results="show"}

ggplot(data=ribo_uct %>%
    inner_join(scores) %>%
    filter(uATGCt == 1) %>%
        left_join(ctbl_big) %>%
        mutate(u1.posTSS20 =  cut(u1.posTSS,breaks=c(0,20,2000))),
       aes(x=u1.scorekw,y=TE)) +
    facet_wrap(~u1.posTSS20) + 
    geom_point(colour="grey50",size=0.7) +
    stat_smooth(method="lm") + 
    scale_y_log10nice("TE of aORF",
                      limits=c(0.05,10),
                      oob=scales::squish) +
    labs(x="uATG wide score")
```


## Compare aATG and dATG context by gene

### Most dATG scores are less than aATG scores

```{r dvsaATG_scoresdens, dependson="calc_scores", fig.width=3.5,fig.height=1.8}

ggplot(data=scores) +
    geom_vline(xintercept=0,size=0.3) +
    geom_density(aes(x=d1vsan,colour="narrow"),kernel="rectangular") +
    # geom_density(aes(x=d1vsaw,colour="wide"),kernel="rectangular")  +
    labs(x="difference in score, dATG vs aATG",
         colour="context") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank(),
          legend.position=c(0.8,0.8))
```

### For highly translated genes, most dATG narrow scores are much less than aATG

```{r dvsaATG_scoresn, dependson="calc_scores", fig.width=3.5,fig.height=3.5}
highdiffn_dvsaATG <- scores %>%
    arrange(desc(d1vsan)) %>%
    head(n=400) %>%
    select(Gene,aATG.scorekn,d1.scorekn)

ggplot(data=scores %>% filter(!is.na(d1.scorekn)),
       aes(x=aATG.scorekn,y=d1.scorekn)) +
    geom_diagline() +
    geom_point(size=0.5,colour="grey50") +
    geom_point(data=highdiffn_dvsaATG,
               colour="red",size=1) +
    geom_point(data=scores %>% 
                   filter(Gene %in% hiTrans),
               colour="blue",size=1) +
    geom_point(data=highdiffn_dvsaATG %>% 
                   filter(Gene %in% hiTrans),
               colour="purple4",size=1.3) +
    labs(x="aATG score, narrow",
         y="dATG score, narrow")
```

Red: high dATG vs aATG **Kozak** score.
Blue: highly translated.
Purple: both.

### Small negative correlation between dATG and aATG score narrow

R = `r with(scores, cor(aATG.scorekn,d1.scorekn,use="pairwise.complete.obs")) %>% round3`

### For highly translated genes, most dATG wide scores are much less than aATG

```{r dvsaATG_scoresw, dependson="calc_scores",fig.width=3.5,fig.height=3.5}
highdiffw_dvsaATG <- scores %>%
    arrange(desc(d1vsaw)) %>%
    head(n=400) %>%
    select(Gene,aATG.scorekw,d1.scorekw)

ggplot(data=scores %>% filter(!is.na(d1.scorekw)),
       aes(x=aATG.scorekw,y=d1.scorekw)) +
    geom_diagline() +
    geom_point(size=0.5,colour="grey50") +
    geom_point(data=highdiffw_dvsaATG,
               colour="red",size=1) +
    geom_point(data=scores %>%
                   filter(Gene %in% hiTrans),
               colour="blue",size=1) +
    geom_point(data=highdiffw_dvsaATG %>%
                   filter(Gene %in% hiTrans),
               colour="purple4",size=1.3) +
    labs(x="aATG score, wide",
         y="dATG score, wide")
```

Red: high dATG vs aATG **Kozak** score.
Blue: highly translated.
Purple: both.

### Small negative correlation between dATG and aATG score wide

R = `r with(scores, cor(aATG.scorekw,d1.scorekw,use="pairwise.complete.obs")) %>% round3`


## Genes with unusual dATG vs aATG narrow score

Those genes are in this list:

```{r highdiffn_dvsaATG,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG
```

* NCU05040, hypothetical conserved protein with GFA Glutathione-dependent formaldehyde-activating enzyme domain. JEC21 homolog is CNC03895
* NCU02127 methylcrotonyl carboxylase-2, mcc-2
* NCU05555, hypothetical protein
* NCU07312, hypothetical protein sgr-24
* NCU12086, hypothetical protein
* NCU02616, WD-repeat containing protein slp1/cdc20
* NCU01333, Delta(24(24(1)))-sterol reductase erg-4, ER localized in Scer
* NCU03699, zinc finger transcription factor-13, znf-13
* NCU05306, hypothetical protein, homolog of Scer DAN4, CNJ01800 and SPBC725.03 (pyridoxamine 5'-phosphate oxidase, cell wall mannoprotein)
* NCU06558 D-amino-acid oxidase oxD, many Crypto homologs (CNAG_02532/CNAG_05802/CNAG_03562/) Sp dao1.


Some of these are well translated:

```{r highdiffn_dvsaATG_hiTrans,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene, d1.frame,d1.posATG) ) %>% 
    filter(Gene %in% hiTrans) 
```

* NCU00405 glycyl-tRNA synthetase 1
* NCU01418 clock-controlled gene-6, ccg-6, SPI1 homolog
* NCU09560 superoxide dismutase-3 sod-3
* NCU01221 RPL16/crp-46, check annotation, suspect N-terminal isn't actually there as not homologous to RPL16 or structured.
* NCU08502 RPS6/crp-14
* NCU04510 aldose reductase gcy-3, Scer homologs YPR1/GCY1 (both cyto/nuc), GRE3, YJR096W 
* NCU09331 HMF1, homolog of Scer HMF1 (cyto) / MMF1 (mito) *which have the same aATG/aATG architecture in S. pombe*
* NCU00629 6-phosphofructokinase emp-3
* NCU01981 translation factor SUI1/eIF1
* NCU07408 po / RPP0


### dATG in frame with ATG

Files with high difference in narrow score, filtered for reasonable amounts of RNA translation, in frame. Saved to *dvsaATG_highdiffn_inframe_Nc.txt*.

```{r highdiffn_dvsaATG_inframe,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG_enoughinframe <- 
    highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene,d1.frame) ) %>%
    filter(Gene %in% enoughRNA, d1.frame==0) %>%
    select(Gene,aATG.scorekn,d1.scorekn)

highdiffn_dvsaATG_enoughinframe

cat(
    "# dvsaATG_highdiffn_inframe_Nc.txt
# Genes with in-frame high-score dATGs
# In N. crassa:
#   - in top 5% by difference, aATG.scorekn - d1.scorekn
#   - in top 40% by RNA abundance
#   - dATG is in frame with aATG
#
# Gene: systematic gene name
# aATG.scorekn: annotated ATG score against narrow Kozak consensus sequence
# d1.scorekn: 1st downstream ATG score against narrow Kozak consensus sequence
#
# Edward Wallace, Jan 2019
",
file="data_out/dvsaATG_highdiffn_inframe_Nc.txt")
highdiffn_dvsaATG_enoughinframe %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/dvsaATG_highdiffn_inframe_Nc.txt",append = TRUE,col_names=TRUE)
```


### dATG out of frame with ATG

Files with high difference in narrow score, filtered for reasonable amounts of RNA translation, out of frame. Saved to *dvsaATG_highdiffn_outframe_Nc.txt*.


```{r highdiffn_dvsaATG_outframe,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG_enoughoutframe <- 
    highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene,d1.frame) ) %>%
    filter(Gene %in% enoughRNA, d1.frame!=0) %>%
    select(Gene,aATG.scorekn,d1.scorekn)

highdiffn_dvsaATG_enoughoutframe

cat(
    "# dvsaATG_highdiffn_outframe_Nc.txt
# Genes with in-frame high-score dATGs
# In N. crassa:
#   - in top 5% by difference, aATG.scorekn - d1.scorekn
#   - in top 40% by RNA abundance
#   - dATG is out-of-frame with aATG
#
# Gene: systematic gene name
# aATG.scorekn: annotated ATG score against narrow Kozak consensus sequence
# d1.scorekn: 1st downstream ATG score against narrow Kozak consensus sequence
#
# Edward Wallace, Jan 2019
",
file="data_out/dvsaATG_highdiffn_outframe_Nc.txt")
highdiffn_dvsaATG_enoughoutframe %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/dvsaATG_highdiffn_outframe_Nc.txt",append = TRUE,col_names=TRUE)
```


### dATG in frame with ATG, wide

Files with high difference in wide score, filtered for top 40% by RNA abundance, in frame. Saved to *dvsaATG_highdiffw_inframe_Nc.txt*.

```{r highdiffw_dvsaATG_inframe,dependson="dvsaATG_scoresn",results="show"}
highdiffw_dvsaATG_enoughinframe <- 
    highdiffw_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene,d1.frame) ) %>%
    filter(Gene %in% enoughRNA, d1.frame==0) %>%
    select(Gene,aATG.scorekw,d1.scorekw)

highdiffw_dvsaATG_enoughinframe

cat(
    "# dvsaATG_highdiffw_inframe_Nc.txt
# Genes with in-frame high-score dATGs
# In N. crassa:
#   - in top 5% by difference, aATG.scorekw - d1.scorekw
#   - in top 40% by RNA abundance
#   - dATG is in frame with aATG
#
# Gene: systematic gene name
# aATG.scorekw: annotated ATG score against wide Kozak consensus sequence
# d1.scorekw: 1st downstream ATG score against wide Kozak consensus sequence
#
# Edward Wallace, Jan 2019
",
file="data_out/dvsaATG_highdiffw_inframe_Nc.txt")
highdiffw_dvsaATG_enoughinframe %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/dvsaATG_highdiffw_inframe_Nc.txt",append = TRUE,col_names=TRUE)
```


### dATG out of frame with ATG, wide

Files with high difference in wide score, filtered for top 40% by RNA abundance, out of frame. Saved to *dvsaATG_highdiffw_outframe_Nc.txt*.


```{r highdiffw_dvsaATG_outframe,dependson="dvsaATG_scoresn",results="show"}
highdiffw_dvsaATG_enoughoutframe <- 
    highdiffw_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene,d1.frame) ) %>%
    filter(Gene %in% enoughRNA, d1.frame!=0) %>%
    select(Gene,aATG.scorekw,d1.scorekw)

highdiffw_dvsaATG_enoughoutframe

cat(
    "# dvsaATG_highdiffw_outframe_Nc.txt
# Genes with in-frame high-score dATGs
# In N. crassa:
#   - in top 5% by difference, aATG.scorekw - d1.scorekw
#   - in top 40% by RNA abundance
#   - dATG is out-of-frame with aATG
#
# Gene: systematic gene name
# aATG.scorekw: annotated ATG score against wide Kozak consensus sequence
# d1.scorekw: 1st downstream ATG score against wide Kozak consensus sequence
#
# Edward Wallace, Jan 2019
",
file="data_out/dvsaATG_highdiffw_outframe_Nc.txt")
highdiffw_dvsaATG_enoughoutframe %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/dvsaATG_highdiffw_outframe_Nc.txt",append = TRUE,col_names=TRUE)
```


## Compare score difference to localization predictions


### Load predictions from mitofates

In input file *NC12_mitofates.txt*.

```{r load_mitofates}
mitofates <- read_mitofates("data_in/NC12_mitofates.txt")

mitocounts <- 
    mitofates %>% 
    group_by(Pred_preseq) %>% 
    count() 
mitobreaks <- mitocounts %>%
    mutate(label=paste0(Pred_preseq," (n = ",n,")")) %>%
    .$label %>%
    set_names(mitocounts$Pred_preseq)

mitopredlist <- mitofates %>%
    filter(Pred_preseq =="Yes") %>%
    .$Gene
```


### Nc Genes with high dATG vs aATG score *are* enriched in mitochondrial presequences

```{r dvsaATG_scoresdiffn_mitofates, dependson=c("calc_scores","load_mitofates"), fig.width=5,fig.height=1.8}

ggplot(data=scores %>% 
           filter(!is.na(d1.scorekn)) %>%
           left_join( mitofates %>% 
                          select(Gene,Prob_preseq,Pred_preseq)
                      ) ,
       aes(x=d1.scorekn-aATG.scorekn,colour=Pred_preseq)) +
    geom_vline(xintercept=0,size=0.3) +
    geom_density(kernel="rectangular",bw=0.025) +
    scale_colour_manual(values=c("Yes"="magenta","No"="grey50"),
                        labels=mitobreaks)  + 
    labs(x="Kozak score, difference dATG - aATG", colour="Mito. pred") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank())
```

### Spectacular correlations between d1 score, d1 frame, and localization

```{r dvsaATG_scoresdiffbar_mitofates, dependson=c("calc_scores","load_mitofates"),fig.width=4.5,fig.height=4}

ggplot(data=scores %>% 
           filter(!is.na(d1.scorekn), !is.na(d1vsaw)) %>%
           left_join(ctbl_big) %>% 
           mutate(d1vsawfac = cut(d1vsaw,
                                  breaks=c(-0.5,0.1,0.5),
                                  labels=c("ATG score\nd < a + 0.1",
                                           "ATG score\na + 0.1 < d")),
                  d1.framefac = factor(d1.frame==0,
                                       levels=c(TRUE,FALSE),
                                       labels=c("In","Out")),
                  enoughRNA = factor(Gene %in% enoughRNA,
                                       levels=c(TRUE,FALSE),
                                       labels=c("enough RNA","not enough"))
                  ) %>%
           inner_join( mitofates %>% 
                          select(Gene,Prob_preseq,Pred_preseq)
                      ) ,
       aes(x=d1.framefac,fill=Pred_preseq)) +
    geom_bar() +
    scale_y_continuous(expand=c(0,0)) + 
    facet_wrap(~ enoughRNA + d1vsawfac,ncol=2,scales="free") + 
    # facet_grid(enoughRNA ~ d1vsawfac,scales="free") + 
    scale_fill_manual(values=c("Yes"="magenta","No"="grey50"),
                        labels=mitobreaks) +
    labs(x="dATG frame", y="number of genes",fill="Mito. pred.") +
    panel_border()
```

### Count dATG vs ATG score, d1 frame, mito pre, enough RNA

```{r dvsaATG_framemitosumm, dependson=c("calc_scores","load_mitofates"),results="show",fig.width=2.5,fig.height=2}
dvsaATG_framemitosumm <- 
    calc_dvsaATG_framemitosumm(scores,ctbl_big,
                               enoughRNA,mitofates)

dvsaATG_framemitosumm

write_tsv(dvsaATG_framemitosumm,
          "data_out/dvsaATG_framemitosumm_Nc.txt")
```

### Nc mito-localized genes also do not have a distinctive aATG context

Although the +5 T is striking.

```{r aATG_motif_mitofates, dependson=c("calc_scores","load_mitofates"), fig.width=4,fig.height=1}

ggseqlogo(data=ctbl_big %>%
              filter(Gene %in% mitopredlist) %>%
              .$aATG.context,
          namespace="TCAG")  + 
    theme_nothing()

```

## Genes with unusual dATG vs aATG narrow score AND predicted mito loc

Those genes are in this list:

```{r highdiffn_dvsaATG_mitopred,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene, d1.frame,d1.posATG) ) %>%
    left_join( ribosum ) %>%
    filter(Gene %in% mitopredlist) %>%
    print(n=26)
```

Most have in-frame ATG

* NCU05040 again
* NCU02127 again
* NCU08568 adenine-19, ad-19, Scer homolog YSA1 is dual-localized.
* NCU03605 amidohydrolase
* NCU10044 hypothetical protein apn-1, AP endonuclease, Sc APN1 dula-localized mito/nuc.
* NCU06615 hypothetical protein, CNAG_06355 homolog, bacterial homologs (YccU in E coli), CoA binding domain, likely mitochondrial.
* NCU01307 hypothetical protein
* NCU03859 MYG1 protein, homologous to mammalian MYG1/Gamm1 which is dual mito/nuc localized, and CNAG_06946 which has same structure.
* NCU01784 pseudouridylate synthase 3 psu-3, tRNA-PUS, DEG1 homolog
* NCU07528 tRNA-pseudouridine synthase-1, tpus-1 dual-loc conserved
* NCU03054 hypothetical protein, paralog of NCU08606, ortholog of ACK1 (mito), SHC1 (?), DSF2 (bud tip)
* NCU03793 hypothetical protein, CNAG_07308/CNA00310 homolog, aflatoxin biogenesis/HypA related.
* NCU08195 ArgRS
* NCU00405 GlyRS
* NCU00742 glycerol-3-phosphate dehydrogenase glp-1, Scer homologs GPD1/GPD2 dual-localized.
* NCU11195 D-isomer specific 2-hydroxyacid dehydrogenase
* NCU00847 hypothetical protein, putative methyltransferase/methoxylase/GA4 desaturase family protein
* NCU03696 nitroreductase, Scer HBN1 & FRM2 homolog, CNAG_02692 (mito) homolog.
* NCU03339 Glutathione reductase gtr-1, GLR1 homolog Cytosolic and mitochondrial glutathione reductase.

### Some more decently-translated ones

```{r highdiffn_dvsaATG_mitopred_hiTrans,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene, d1.frame,d1.posATG) ) %>%
    left_join( ribosum ) %>%
    filter(Gene %in% mitopredlist, Gene %in% midTrans) 
```

* NCU03749 pyruvate-1 pru-1, GLO2/GLO4 homolog
* NCU06877 CRAL/TRIO domain-containing protein, CSR1 homolog dual-localized.
* NCU06108 UPF0135, NIF3 homolog
* NCU05805 Serine hydroxymethyltransferase cbs-2, SHM1 (mito) and SHM2 (cyto) homolog.
* NCU01501 mitochondrial peptidyl-tRNA hydrolase Pth2
* NCU03177 sco1, Scer homologs are mitochondrial, only 2 codons between ATGs.
* NCU01965 ValRS
* NCU09536 conserved hypothetical protein
* NCU02701 dipeptidyl peptidase dpp-2, YOL057W homolog
* NCU00936 succinate-3 suc-3; Cneo has homologs CNAG_01027 (mito) and CNAG_04467 (not); Scer has a single homolog UGA2, mostly cytoplasmic

<a href="#top">Back to table of contents</a>
