---
title: "Start codon context for Schizosaccharomyces pombe"
author: Edward Wallace, ewjwallace@gmail.com
date: 16 Jan 2019
output:
  # html_notebook: default
  html_document:
    toc: true
    toc_depth: 4
---

# Summary

This presents the analysis of Schizosaccharomyces start codon usage and context. This uses the S. pombe 972h- translation estimates from Duncan and Mata 2017.

### Load Packages

```{r setup,message=FALSE,echo=FALSE,cache=FALSE}
knitr::opts_chunk$set(warning=FALSE,message=FALSE,echo=FALSE,cache=TRUE,
                      results="hide",
                      fig.path="figure/SpombeATGContext-",
                      cache.path="cache/SpombeATGContext-")
# load common functions
source("../SharedScripts/sharedScripts.R")
# setwd("~/Repos/CryptoTranscriptome2018/ATGContextOthers/")
```


# Here we go

## Expression: RNA abundance and ribosome-protected-fragments

### Load expression data

We remove the Sp mitochondrial genes here because their translation is not detected by this ribosome profiling protocol.

```{r load_ribosum,results="show"}

# tpms <- read_tsv("data_in/ExpressionTranslation/SpombeDuncan2017_TPMs_collated.tsv") 
# ribosum <- tpms %>%
#     mutate(RPF=(NCHX1+NCHX3+NnoC1+NnoC3)/4,
#            RPF_noN=(SCHX1+SCHX3+SnoC1+SnoC3)/4) %>%
#     select(Gene=ORF,RPF,RPF_noN)

cds_lengths <- read_tsv("data_in/ExpressionTranslation/Spombe_DuncanMata2017_cds_lengths.txt",
                   comment="#",skip=1,col_names = c("Gene","CDSlength"))
counts <- read_tsv("data_in/ExpressionTranslation/Spombe_DuncanMata2017_readcounts.txt",
                   comment="#") 

RDenss <- counts %>%
    gather("Sample","Count",-Gene) %>% 
    separate(Sample,into = c("Strain","Condition","Drug","Type","Region","Rep"),
             remove=TRUE) %>%
    mutate(Sample = paste(Type,Condition,Drug,Rep,sep="_")) %>%
    left_join(cds_lengths,by="Gene") %>%
    mutate(RDens=Count/CDSlength) %>%
    select(Gene,Sample,Type,Condition,Drug,Rep,Count,RDens)

TPMs_wide <- RDenss %>%
    group_by(Sample) %>%
    summarize(RDens_total=sum(RDens)) %>%
    right_join(RDenss) %>%
    mutate(TPM = RDens * 1e6 / RDens_total) %>%
    select(Gene,Sample,TPM) %>%
    spread(Sample,TPM) 

ribosum <- TPMs_wide %>%
    filter(!str_detect(Gene,"SPMIT")) %>%
    mutate(RNA = (mrna_N_CHX_1+mrna_N_CHX_3+mrna_N_noCHX_1+mrna_N_noCHX_3)/4,
           RNA_noN = (mrna_noN_CHX_1+mrna_noN_CHX_3+mrna_noN_noCHX_1+mrna_noN_noCHX_3)/4,
           RPF = (ribo_N_CHX_1+ribo_N_CHX_3+ribo_N_noCHX_1+ribo_N_noCHX_3)/4,
           RPF_noN = (ribo_noN_CHX_1+ribo_noN_CHX_3+ribo_noN_noCHX_1+ribo_noN_noCHX_3)/4
           ) %>%
    select(Gene,RNA,RPF,RNA_noN,RPF_noN) %>%
    mutate(TE=RPF/RNA,TE_noN=RPF_noN/RNA_noN)

ribosum %>%
    arrange(desc(RPF)) 

hiTrans <- ribosum %>%
    arrange(desc(RPF)) %>%
    head(n=256) %>%
    .$Gene

midTrans <- ribosum %>%
    arrange(desc(RPF)) %>%
    head(n=1280) %>%
    .$Gene

nonterribleTrans <- ribosum %>%
    arrange(desc(RPF)) %>%
    head(n=3840) %>%
    .$Gene

enoughRNA <- ribosum %>%
    arrange(desc(RNA)) %>%
    head(n=2560) %>%
    .$Gene
```

#### Check replicability

```{r scatter_ribo,dependson="load_ribosum",results="show",fig.width=10,fig.height=10}
library(GGally)

ggpairs(data=TPMs_wide %>% select(-Gene) %>% log10())
```

### Ribosome occupancy mostly tracks RNA abundance

```{r plot_TE,dependson="load_ribosum",fig.height=3.9,fig.width=4.1}
TE_unusual <- 
    ribosum %>%
    filter( (RPF > 1.5e4) |
                (RPF > 1000 & RPF > RNA * 10 ) | 
                (RNA > 400 & RPF < RNA / 10 ))


ggplot(data=ribosum,aes(x=RNA,y=RPF)) +
    geom_diagline() +
    geom_point(size=.8,colour="grey50") +
    geom_point(data=TE_unusual,aes(colour=Gene),size=1) + 
    geom_text_repel(data=TE_unusual,aes(label=Gene,colour=Gene)) +
    scale_loglog(limits=c(1e-1,2e4),
                  oob=scales::squish,
                  expand=c(0.01,0.01)) +
    labs(x="RNA abundance (TPM of aORF)",
         y="Ribosome occupancy (TPM of aORF)") +
    theme(legend.position="none")
```


## ATG Context

### Load context data

```{r load_context,results="show"}
ctbl_big <- read_ATGcontext_table("data_in/SPombe_allATGcontext.table.txt")
ctbl_big
```

### Some genes are not in both context and ribosum data

```{r mismatched_genes,dependson=c("load_ribosum","load_context"),results="show"}
setdiff(ctbl_big$Gene,ribosum$Gene)
setdiff(ribosum$Gene,ctbl_big$Gene)
```


### Annotated ATGs have a Kozak consensus sequence

```{r aATG_context,fig.width=4,fig.height=1}
ggseqlogo(data=ctbl_big$aATG.context,namespace="TCAG") + 
    theme_nothing()
```

### Highly translated Annotated ATGs have a Kozak consensus sequence

That's for hiTrans, the top 5% (256) translated genes by RPF TPM.

```{r a_hi_context,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big,Gene %in% hiTrans)$aATG.context,
                      namespace="TCAG") + 
    theme_nothing()
```

### Cytoplasmic Ribosome Annotated ATGs have a Kozak consensus sequence

```{r cytoribo_context,dependson="load_context",fig.width=4,fig.height=1}
cytoribolist <- read_tsv("data_out/ComplexOrthologs/cytoribosome_PTRcurated6.txt",
                        comment="#") %>%
    filter(SPECIES=="SCHPO") %>%
    .$GENE_ID
ggseqlogo(data=filter(ctbl_big,Gene %in% cytoribolist)$aATG.context,
                      namespace="TCAG") + 
    theme_nothing()
```

### Most cytoribo genes are highly translated (5% highest)

Venn diagram.

There are many duplicate cytoRibo genes in Sp.

```{r hitrans_cytoribo_venn,dependson=c("cytoribo_context","load_ribosum"),fig.width=2.1,fig.height=2}
venn(hiTrans=hiTrans,cytoRibo=cytoribolist)
```


### Downstream ATGs don't have a consensus

First downstream ATG

```{r d1_context,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big,nchar(d1.context) ==24)$d1.context,
                      namespace="TCAG") + 
    theme_nothing()
```

### Downstream ATGs in frame and highly translated don't have a consensus

Except for 3rd-codon-position bias.

```{r d1_hi_frame0_context,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big,
                      nchar(d1.context) == 24,
                      d1.frame==0,
                      Gene %in% hiTrans)$d1.context,
                      namespace="TCAG") + 
    theme_nothing()
```

## Calculate Information content and scores of consensus motif

### Calculate a wide and a narrow consensus sequence

Calculate motif score against the position weight matrix (pwm) for both **narrow** (-4 from ATG through to ATG) and **wide** (-10 from ATG) kozak consensus motif. These motifs are taken from the top 5% highly translated genes.

```{r kozak_consensus,fig.width=4,fig.height=1}
kozak_all_PFM <- filter(ctbl_big,Gene %in% hiTrans) %>%
    .$aATG.context %>%
    
    str_to_upper() %>%
    consensusMatrix() 

kozak_w_PWM <- PWM(kozak_all_PFM[,3:15])
kozak_n_PWM <- PWM(kozak_all_PFM[,9:15])
```


```{r write_PFM, dependson="kozak_consensus",results="show"}
cat(
    "# PFM_hiTrans_Sp.txt
# Position Frequency Matrix for aATGs of 256 highest-translated genes from Sp
# This is of the wide ATG context NNNNNNNNNNNNATGNNNNNNNNN
# Column is position, Row is nt count in order: ACGT.
#
# Edward Wallace, June 2018
",
file="data_out/PFM_hiTrans_Sp.txt")
kozak_all_PFM %>%
    as.tibble() %>%
    write_tsv("data_out/PFM_hiTrans_Sp.txt",
              append = TRUE,col_names=FALSE)
```

### Estimate the information content 

Using the sequence logo, details on https://en.wikipedia.org/wiki/Sequence_logo

```{r kozak_inf,dependson=c("kozak_consensus","cytoribo_context"),fig.width=4,fig.height=1,results="show"}
inf_content_narrow <- 
    tribble(
        ~Genes, ~ATG, ~Width, ~Infon,
        "All", "aATG", "narrow", 
        ctbl_big %>%
            .$aATG.context %>%
            infonfromseqs(sstart=9L,send=12L),
        "HiTrans", "aATG", "narrow", 
        ctbl_big %>%
            filter(Gene %in% hiTrans) %>% 
            .$aATG.context %>%
            infonfromseqs(sstart=9L,send=12L),
        "CytoRibo", "aATG", "narrow", 
        ctbl_big %>%
            filter(Gene %in% cytoribolist) %>% 
            .$aATG.context %>%
            infonfromseqs(sstart=9L,send=12L),
        "All", "d1ATG", "narrow", 
        ctbl_big %>%
            .$d1.context %>%
            na.omit() %>%
            infonfromseqs(sstart=9L,send=12L),
        "HiTrans", "d1ATG", "narrow", 
        ctbl_big %>%
            filter(Gene %in% hiTrans) %>% 
            .$d1.context %>%
            na.omit() %>%
            infonfromseqs(sstart=9L,send=12L),
        "CytoRibo", "d1ATG", "narrow", 
        ctbl_big %>%
            filter(Gene %in% cytoribolist) %>% 
            .$d1.context %>%
            na.omit() %>%
            infonfromseqs(sstart=9L,send=12L)
    )

write_tsv(inf_content_narrow %>%
              mutate_if(is.numeric,round3),
          "data_out/inf_content_narrow_Sp.txt")
inf_content_narrow
```

Information content in bits of highly-translated consensus (excluding 6 bits from ATG), **narrow** is `r infontot(kozak_all_PFM[,5:9]) %>% round(2)`, of **wide** is `r infontot(kozak_all_PFM[,-(10:12)]) %>% round(2)`.


### Estimate information content per base across start context

```{r kozak_inf_perbase,dependson=c("kozak_consensus","cytoribo_context"),fig.width=6,fig.height=1.4,results="show"}
inf_content_perbase <- 
    tibble(i = 1:24, 
           Pos = c(-12:-1, 1:12),
        All = ctbl_big %>%
            .$aATG.context %>%
            infonfromseqs(scope="all"),
        HiTrans = ctbl_big %>%
            filter(Gene %in% hiTrans) %>% 
            .$aATG.context %>%
            infonfromseqs(scope="all"),
        CytoRibo = ctbl_big %>%
            filter(Gene %in% cytoribolist) %>% 
            .$aATG.context %>%
            infonfromseqs(scope="all")
    )

write_tsv(inf_content_perbase %>%
              mutate_if(is.numeric,round3),
          "data_out/inf_content_perbase_Sp.txt")
# inf_content_perbase

plot_inf_perbase(inf_content_perbase)
```

### Calculate scores of aATG, dATG, uATG against Kozak consensus

```{r calc_scores, dependson="kozak_consensus"}
scores <- 
    ctbl_big %>%
    transmute(Gene,
              aATG.scorekn = PWMscoren(str_to_upper(aATG.context),kozak_n_PWM),
              d1.scorekn   = PWMscoren(d1.context,kozak_n_PWM),
              u1.scorekn   = PWMscoren(str_to_upper(u1.context),kozak_n_PWM),
              aATG.scorekw = PWMscorew(str_to_upper(aATG.context),kozak_w_PWM),
              d1.scorekw   = PWMscorew(d1.context,kozak_w_PWM),
              u1.scorekw   = PWMscorew(str_to_upper(u1.context),kozak_w_PWM),
              NULL
    ) %>%
    mutate(d1vsan = d1.scorekn - aATG.scorekn,
           u1vsan = u1.scorekn - aATG.scorekn,
           d1vsaw = d1.scorekw - aATG.scorekw,
           u1vsaw = u1.scorekw - aATG.scorekw,
           NULL) 


# ribotbl_scoresenough <-
#     ribosum %>%
#     filter(Gene %in% enoughRNA) %>%
#     # left_join(ribotbl,by="Gene") %>% 
#     left_join(scores,by="Gene")

```

We calculate scores using Biostrings::PWMscoreStartingAt.

The best description I could find of this method is: https://support.bioconductor.org/p/61520/

It is just the sum of the matrix product of the PWM with the sequence.


Write scores to file *scores_kozak_Sp.txt*.

```{r write_scores, dependson="calc_scores",results="show"}
scores
cat(
    "# scores_kozak_Sp.txt
# Scores of ATG context against Kozak sequence, S. pombe 972h-
# Kozak sequence derived from 256 highest-translated genes by RPF
#
# Gene: Sp systematic gene name
# aATG.scorekn: score of annotated ATG against narrow Kozak sequence (NNNNATG)
# d1.scorekn: score of 1st downstream ATG against narrow
# xx.scorekw: analogous scores against wide Kozak sequence (NNNNNNNNNNATG)
#
# Edward Wallace, June 2018
",
file="data_out/scores_kozak_Sp.txt")
scores %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/scores_kozak_Sp.txt",
              append = TRUE,col_names=TRUE)
```

### Plot against narrow consensus (-4 to ATG)

```{r plot_scores_n, dependson="calc_scores", fig.width=5,fig.height=1.8}

sbandwidthn = 0.025

ggplot(data=scores) +
    geom_density(aes(x=aATG.scorekn,colour="aATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    geom_density(aes(x=d1.scorekn,colour="dATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    # geom_density(aes(x=u1.scorekn,colour="uATG"),
    #             kernel="rectangular",bw=sbandwidthn) +
    geom_density(data=filter(scores,Gene %in% hiTrans),
                 aes(x=aATG.scorekn,colour="aATG, top 5%"),
                 kernel="rectangular",bw=sbandwidthn) +
    scale_colour_manual(values=c("aATG"="dodgerblue",
                                 "aATG, top 5%"="darkblue",
                                 # "uATG"="red",
                                 "dATG"="green3") ) +
    labs(x="ATG narrow Kozak score, Sp") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank(),
          legend.title=element_blank())
```

### Plot against wide consensus (-10 to ATG)

```{r plot_scores_w, dependson="calc_scores", fig.width=5,fig.height=1.8}

sbandwidthn = 0.025

ggplot(data=scores) +
    geom_density(aes(x=aATG.scorekw,colour="aATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    geom_density(aes(x=d1.scorekw,colour="dATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    geom_density(aes(x=u1.scorekw,colour="uATG"),
                kernel="rectangular",bw=sbandwidthn) +
    geom_density(data=filter(scores,Gene %in% hiTrans),
                 aes(x=aATG.scorekw,colour="aATG, top 5%"),
                 kernel="rectangular",bw=sbandwidthn) +
    scale_colour_manual(values=c("aATG"="dodgerblue",
                                 "aATG, top 5%"="darkblue",
                                 "uATG"="red",
                                 "dATG"="green3") ) +
    labs(x="ATG wide Kozak score, Sp") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank(),
          legend.title=element_blank())
```


## Mutual information of different positions around ATG

### For all annotated genes

```{r mutinf,dependson=c("load_context","load_riboviz"),results="show",fig.height=4,fig.width=5}
mutinf <- mutinfostr(ctbl_big$aATG.context)
# mutinf

plot_mutinfo(mutinf)
```

### For highly translated genes

```{r mutinf_hiTrans,dependson=c("load_context","load_riboviz"),results="show",fig.height=4,fig.width=5}
ctbl_big %>%
    filter(Gene %in% hiTrans) %>%
    .$aATG.context %>%
    mutinfostr %>%
    plot_mutinfo
```


### For cytoplasmic ribosomal proteins

```{r mutinf_cytoribo,dependson=c("load_context","load_riboviz"),results="show",fig.height=4,fig.width=5}
ctbl_big %>%
    filter(Gene %in% cytoribolist) %>%
    .$aATG.context %>%
    mutinfostr %>%
    plot_mutinfo

```


### What is the -2 correlation in cyto ribo genes?

```{r cytoribo_contextm2,dependson="load_context",fig.width=4,fig.height=3}

seqlogoFilterGenePos <- function(datain=ctbl_big,Genelist=hiTrans,
                                 pos=8L, nt="T") {
    datafiltered <- filter(datain,
                          Gene %in% Genelist,
                          str_sub(aATG.context,start=pos,end=pos)==nt
    )
    ggseqlogo(data=datafiltered$aATG.context,
    namespace="TCAG") + 
        theme_nothing() +
        annotate(geom="text",x=21,y=1,
                 label=paste0("n = ", nrow(datafiltered)))
}

plot_grid(seqlogoFilterGenePos(Genelist=cytoribolist,pos=11L,nt="T"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=11L,nt="C"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=11L,nt="A"),
          seqlogoFilterGenePos(Genelist=cytoribolist,pos=11L,nt="G"),
    ncol=1
)
    
```

There are definitely correlations there.


## uATGs inhibit translation of the main ORF

### uATGs are associated with lower absolute translation


```{r uATG_RPF, dependson="calc_scores", fig.width=2.5,fig.height=3.5}
uATG_ct <- read_tsv("data_in/SPombe_allATGcontext.table.txt") %>% 
    dplyr::rename(Gene=gene) %>%
    dplyr::rename_all(str_replace,pattern="-ATG",replacement="") %>%
    dplyr::select(c("Gene",paste0("u",1:56,".posTSS"))) %>%
    gather(key="uATG",value="posTSS",-Gene) %>%
    mutate(posTSS=as.integer(posTSS)) %>%
    group_by(Gene) %>%
    summarize(uATGCt = sum(!is.na(posTSS)),
              uATGCtmin20 = sum(posTSS >= 20,na.rm=TRUE)
              ) %>%
    mutate(uATGCtC = replace(uATGCt,uATGCt>=2, "2+") %>%
               factor(levels=c(0:1,"2+")),
           uATGCtmin20C = replace(uATGCtmin20,uATGCtmin20>=2, "2+") %>%
               factor(levels=c(0:1,"2+")),
           uATGfac = make_uATGfact(uATGCt,uATGCtmin20)
           )

ribo_uct <- 
    ribosum %>%
    filter(Gene %in% enoughRNA) %>%
    inner_join(uATG_ct)

ggplot(data=ribo_uct,
       aes(x=uATGCtC,y=RPF)) +
    geom_jitter(colour="grey50",size=0.7) +
    geom_boxplot(colour="blue",outlier.colour=NA,fill=NA,size=0.75) +
    scale_y_log10nice("Ribosome occupancy (TPM of aORF)",
                      limits=c(1,10000),
                      oob=scales::squish) +
    labs(x="uATG count")
```

### uATGs are associated with lower translation efficiency

```{r uATG_TE, dependson="uATG_RPF", fig.width=2.5,fig.height=3.5}

ggplot(data=ribo_uct,
       aes(x=uATGCtC,y=TE)) +
    geom_jitter(colour="grey50",size=0.7) +
    geom_boxplot(colour="blue",outlier.colour=NA,fill=NA,size=0.75) +
    scale_y_log10nice("TE of aORF",
                      limits=c(0.05,10),expand=c(0,0),
                      oob=scales::squish) +
    labs(x="uATG count")
```

### uATGs associated with lower translation efficiency are over 20nt from TSS

```{r uATGmin20_TE, dependson="uATG_RPF", fig.width=2.5,fig.height=3.5,results="show"}
# ribo_uct
ggplot(data=ribo_uct,
       aes(x=uATGfac,y=TE)) +
    geom_jitter(colour="grey50",size=0.7) +
    geom_boxplot(colour="blue",outlier.colour=NA,fill=NA,size=0.75) +
    scale_y_log10nice("TE of aORF",
                      limits=c(0.05,10),expand=c(0,0),
                      oob=scales::squish) +
    labs(x="uATG count and position\nrelative to TSS")
```

### uATG score does not strongly affect TE

We suspect that uATG is associated with lower TE if the uATG has

* position at least 20nt downstream from TSS
* higher score

This figure shows that, *for genes with only 1 uATG*, this correlation is weak and opposite to expected.

```{r uATG_scoreTE, dependson="uATG_RPF", fig.width=4.5,fig.height=2.5,results="show"}

ggplot(data=ribo_uct %>%
    inner_join(scores) %>%
    filter(uATGCt == 1) %>%
        left_join(ctbl_big) %>%
        mutate(u1.posTSS20 =  cut(u1.posTSS,breaks=c(0,20,1000))),
       aes(x=u1.scorekw,y=TE)) +
    facet_wrap(~u1.posTSS20) + 
    geom_point(colour="grey50",size=0.7) +
    stat_smooth(method="lm") + 
    scale_y_log10nice("TE of aORF",
                      limits=c(0.05,10),
                      oob=scales::squish) +
    labs(x="uATG wide score")
```


## Compare aATG and dATG context by gene


### Most dATG scores are less than aATG scores

```{r dvsaATG_scoresdens, dependson="calc_scores", fig.width=3.5,fig.height=1.8}

ggplot(data=scores) +
    geom_vline(xintercept=0,size=0.3) +
    geom_density(aes(x=d1vsan,colour="narrow"),kernel="rectangular") +
    # geom_density(aes(x=d1vsaw,colour="wide"),kernel="rectangular")  +
    labs(x="difference in score, dATG vs aATG",
         colour="context") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank(),
          legend.position=c(0.8,0.8))
```

### For highly translated genes, most dATG narrow scores are much less than aATG

```{r dvsaATG_scoresn, dependson="calc_scores", fig.width=3.5,fig.height=3.5}
highdiffn_dvsaATG <- scores %>%
    arrange(desc(d1vsan)) %>%
    head(n=400) %>%
    select(Gene,aATG.scorekn,d1.scorekn)

ggplot(data=scores %>% filter(!is.na(d1.scorekn)),
       aes(x=aATG.scorekn,y=d1.scorekn)) +
    geom_diagline() +
    geom_point(size=0.5,colour="grey50") +
    geom_point(data=highdiffn_dvsaATG,
               colour="red",size=1) +
    geom_point(data=scores %>% 
                   filter(Gene %in% hiTrans),
               colour="blue",size=1) +
    geom_point(data=highdiffn_dvsaATG %>% 
                   filter(Gene %in% hiTrans),
               colour="purple4",size=1.3) +
    labs(x="aATG score, narrow",
         y="dATG score, narrow")
```

Red: high dATG vs aATG **Kozak** score.
Blue: highly translated.
Purple: both.

### Small negative correlation between dATG and aATG score narrow

R = `r with(scores, cor(aATG.scorekn,d1.scorekn,use="pairwise.complete.obs")) %>% round3`

### For highly translated genes, most dATG wide scores are much less than aATG

```{r dvsaATG_scoresw, dependson="calc_scores",fig.width=3.5,fig.height=3.5}
highdiffw_dvsaATG <- scores %>%
    arrange(desc(d1vsaw)) %>%
    head(n=400) %>%
    select(Gene,aATG.scorekw,d1.scorekw)

ggplot(data=scores %>% filter(!is.na(d1.scorekw)),
       aes(x=aATG.scorekw,y=d1.scorekw)) +
    geom_diagline() +
    geom_point(size=0.5,colour="grey50") +
    geom_point(data=highdiffw_dvsaATG,
               colour="red",size=1) +
    geom_point(data=scores %>%
                   filter(Gene %in% hiTrans),
               colour="blue",size=1) +
    geom_point(data=highdiffw_dvsaATG %>%
                   filter(Gene %in% hiTrans),
               colour="purple4",size=1.3) +
    labs(x="aATG score, wide",
         y="dATG score, wide")
```

Red: high dATG vs aATG **Kozak** score.
Blue: highly translated.
Purple: both.

### Small negative correlation between dATG and aATG score wide

R = `r with(scores, cor(aATG.scorekw,d1.scorekw,use="pairwise.complete.obs")) %>% round3`


## Genes with unusual dATG vs aATG narrow score

Those genes are in this list:

```{r highdiffn_dvsaATG,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG
```

* SPAC222.04c/Ies6, Ino80 chromatin remodeling complex subunit
* SPAC26F1.05/Mug106, meiotically upregulate, short, no known orthologs.
* SPCC1620.12c GTPase activating protein, distand MDR1/GYP2 homolog
* SPBC9B6.11c CCR4/nocturin family endoribonuclease, NGL2/3 homolog
* SPAC12B10.01c HECT-type ubiquitin-protein ligase E3, UFD4 homolog
* SPAC1B2.06, short, no known orthologs


Some of these are decently translated (top 25%)

```{r highdiffn_dvsaATG_hiTrans,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG %>% 
                   filter(Gene %in% midTrans) 
```

* SPCC31H12.08c CCR4-Not complex subunit Ccr4
* SPBC13G1.01c mitochondrial ribosomal protein subunit S4
* SPCC188.04c NMS complex subunit Spc25
* SPBC1198.08 dipeptidase Dug1
* SPAC1071.01c mRNA cleavage and polyadenylation specificity factor complex subunit Pta1
* SPAC22F3.06c Lon protease homolog Lon1 (mito localized)
* SPBC428.01c nucleoporin Nup107
* SPBC3F6.04c U3 snoRNP protein Nop14

Several things involved in mRNA 3' end regulation. Otheriwse unclear. We should check if those ATGs are actually used.


### dATG in frame with ATG, narrow

Files with high difference in narrow score, filtered for top 50% of RNA, in frame. Saved to *dvsaATG_highdiffn_inframe_Sp.txt*.

```{r highdiffn_dvsaATG_inframe,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG_enoughinframe <- 
    highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene,d1.frame) ) %>%
    filter(Gene %in% enoughRNA, d1.frame==0) %>%
    select(Gene,aATG.scorekn,d1.scorekn)

highdiffn_dvsaATG_enoughinframe

cat(
    "# dvsaATG_highdiffn_inframe_Sp.txt
# Genes with in-frame high-score dATGs
# In S. pombe 972h-:
#   - in top 5% by difference, aATG.scorekn - d1.scorekn
#   - in top 50% by RNA abundance
#   - dATG is in frame with aATG
#
# Gene: systematic gene name
# aATG.scorekn: annotated ATG score against narrow Kozak consensus sequence
# d1.scorekn: 1st downstream ATG score against narrow Kozak consensus sequence
#
# Edward Wallace, Jan 2019
",
file="data_out/dvsaATG_highdiffn_inframe_Sp.txt")
highdiffn_dvsaATG_enoughinframe %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/dvsaATG_highdiffn_inframe_Sp.txt",append = TRUE,col_names=TRUE)
```


### dATG out of frame with ATG, narrow

Files with high difference in narrow score, filtered for top 50% of RNA, out of frame. Saved to *dvsaATG_highdiffn_outframe_Sp.txt*.


```{r highdiffn_dvsaATG_outframe,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG_enoughoutframe <- 
    highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene,d1.frame) ) %>%
    filter(Gene %in% enoughRNA, d1.frame!=0) %>%
    select(Gene,aATG.scorekn,d1.scorekn)

highdiffn_dvsaATG_enoughoutframe

cat(
    "# dvsaATG_highdiffn_outframe_Sp.txt
# Genes with in-frame high-score dATGs
# In S. pombe 972h-:
#   - in top 5% by difference, aATG.scorekn - d1.scorekn
#   - in top 50% by RNA abundance
#   - dATG is out-of-frame with aATG
#
# Gene: systematic gene name
# aATG.scorekn: annotated ATG score against narrow Kozak consensus sequence
# d1.scorekn: 1st downstream ATG score against narrow Kozak consensus sequence
#
# Edward Wallace, Jan 2019
",
file="data_out/dvsaATG_highdiffn_outframe_Sp.txt")
highdiffn_dvsaATG_enoughoutframe %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/dvsaATG_highdiffn_outframe_Sp.txt",append = TRUE,col_names=TRUE)
```


### dATG in frame with ATG, wide

Files with high difference in wide score, filtered for top 50% of RNA, in frame. Saved to *dvsaATG_highdiffw_inframe_Sp.txt*.

```{r highdiffw_dvsaATG_inframe,dependson="dvsaATG_scoresn",results="show"}
highdiffw_dvsaATG_enoughinframe <- 
    highdiffw_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene,d1.frame) ) %>%
    filter(Gene %in% enoughRNA, d1.frame==0) %>%
    select(Gene,aATG.scorekw,d1.scorekw)

highdiffw_dvsaATG_enoughinframe

cat(
    "# dvsaATG_highdiffw_inframe_Sp.txt
# Genes with in-frame high-score dATGs
# In S. pombe 972h-:
#   - in top 5% by difference, aATG.scorekw - d1.scorekw
#   - in top 50% by RNA abundance
#   - dATG is in frame with aATG
#
# Gene: systematic gene name
# aATG.scorekw: annotated ATG score against wide Kozak consensus sequence
# d1.scorekw: 1st downstream ATG score against wide Kozak consensus sequence
#
# Edward Wallace, Jan 2019
",
file="data_out/dvsaATG_highdiffw_inframe_Sp.txt")
highdiffw_dvsaATG_enoughinframe %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/dvsaATG_highdiffw_inframe_Sp.txt",append = TRUE,col_names=TRUE)
```


### dATG out of frame with ATG, wide

Files with high difference in wide score, filtered for top 50% of RNA, out of frame. Saved to *dvsaATG_highdiffw_outframe_Sp.txt*.


```{r highdiffw_dvsaATG_outframe,dependson="dvsaATG_scoresn",results="show"}
highdiffw_dvsaATG_enoughoutframe <- 
    highdiffw_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene,d1.frame) ) %>%
    filter(Gene %in% enoughRNA, d1.frame!=0) %>%
    select(Gene,aATG.scorekw,d1.scorekw)

highdiffw_dvsaATG_enoughoutframe

cat(
    "# dvsaATG_highdiffw_outframe_Sp.txt
# Genes with in-frame high-score dATGs
# In S. pombe 972h-:
#   - in top 5% by difference, aATG.scorekw - d1.scorekw
#   - in top 50% by RNA abundance
#   - dATG is out-of-frame with aATG
#
# Gene: systematic gene name
# aATG.scorekw: annotated ATG score against wide Kozak consensus sequence
# d1.scorekw: 1st downstream ATG score against wide Kozak consensus sequence
#
# Edward Wallace, Jan 2019
",
file="data_out/dvsaATG_highdiffw_outframe_Sp.txt")
highdiffw_dvsaATG_enoughoutframe %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/dvsaATG_highdiffw_outframe_Sp.txt",append = TRUE,col_names=TRUE)
```


## Compare score difference to localization predictions


### Load predictions from mitofates

In input file *SPombe_mitofates.txt*.

```{r load_mitofates}
mitofates <- read_mitofates("data_in/SPombe_mitofates.txt")

mitocounts <- 
    mitofates %>% 
    group_by(Pred_preseq) %>% 
    count() 
mitobreaks <- mitocounts %>%
    mutate(label=paste0(Pred_preseq," (n = ",n,")")) %>%
    .$label %>%
    set_names(mitocounts$Pred_preseq)

mitopredlist <- mitofates %>%
    filter(Pred_preseq =="Yes") %>%
    .$Gene
```


### Sp Genes with high dATG vs aATG score *are* enriched in mitochondrial presequences

```{r dvsaATG_scoresdiffn_mitofates, dependson=c("calc_scores","load_mitofates"), fig.width=5,fig.height=1.8}

ggplot(data=scores %>% 
           filter(!is.na(d1.scorekn)) %>%
           left_join( mitofates %>% 
                          select(Gene,Prob_preseq,Pred_preseq)
                      ) ,
       aes(x=d1.scorekn-aATG.scorekn,colour=Pred_preseq)) +
    geom_vline(xintercept=0,size=0.3) +
    geom_density(kernel="rectangular",bw=0.025) +
    scale_colour_manual(values=c("Yes"="magenta","No"="grey50"),
                        labels=mitobreaks)  + 
    labs(x="Kozak score, difference dATG - aATG", colour="Mito. pred") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank())
```

### Correlations between d1 score, d1 frame, and localization

```{r dvsaATG_scoresdiffbar_mitofates, dependson=c("calc_scores","load_mitofates"),fig.width=4.5,fig.height=4}

ggplot(data=scores %>% 
           filter(!is.na(d1.scorekn), !is.na(d1vsaw)) %>%
           left_join(ctbl_big) %>% 
           mutate(d1vsawfac = cut(d1vsaw,
                                  breaks=c(-0.5,0.1,0.5),
                                  labels=c("ATG score\nd < a + 0.1",
                                           "ATG score\na + 0.1 < d")),
                  d1.framefac = factor(d1.frame==0,
                                       levels=c(TRUE,FALSE),
                                       labels=c("In","Out")),
                  enoughRNA = factor(Gene %in% enoughRNA,
                                       levels=c(TRUE,FALSE),
                                       labels=c("enough RNA","not enough"))
                  ) %>%
           inner_join( mitofates %>% 
                          select(Gene,Prob_preseq,Pred_preseq)
                      ) ,
       aes(x=d1.framefac,fill=Pred_preseq)) +
    geom_bar() +
    scale_y_continuous(expand=c(0,0)) + 
    facet_wrap(~ enoughRNA + d1vsawfac,ncol=2,scales="free") + 
    # facet_grid(enoughRNA ~ d1vsawfac,scales="free") + 
    scale_fill_manual(values=c("Yes"="magenta","No"="grey50"),
                        labels=mitobreaks) +
    labs(x="dATG frame", y="number of genes",fill="Mito. pred.") +
    panel_border()
```

### Count dATG vs ATG score, d1 frame, mito pre, enough RNA

```{r dvsaATG_framemitosumm, dependson=c("calc_scores","load_mitofates"),results="show",fig.width=2.5,fig.height=2}
dvsaATG_framemitosumm <- 
    calc_dvsaATG_framemitosumm(scores,ctbl_big,
                               enoughRNA,mitofates)

dvsaATG_framemitosumm

write_tsv(dvsaATG_framemitosumm,
          "data_out/dvsaATG_framemitosumm_Sp.txt") 
```

### Sp mito-localized genes also do not have a distinctive aATG context

Although the +5 T is striking.

```{r aATG_motif_mitofates, dependson=c("calc_scores","load_mitofates"), fig.width=4,fig.height=1}

ggseqlogo(data=ctbl_big %>%
              filter(Gene %in% mitopredlist) %>%
              .$aATG.context,
          namespace="TCAG")  + 
    theme_nothing()

```

## Genes with unusual dATG vs aATG narrow score AND predicted mito loc

Those genes are in this list:

```{r highdiffn_dvsaATG_mitopred,dependson="dvsaATG_scoresn",results="show"}
highdiffn_dvsaATG %>%
    left_join( ctbl_big %>% select(Gene, d1.frame,d1.posATG) ) %>%
    left_join( ribosum ) %>%
    filter(Gene %in% mitopredlist) %>%
    print(n=26)
```

* SPBC13G1.01c mitochondrial ribosomal protein subunit S4
* SPBC9B6.11c CCR4/nocturin family endoribonuclease, NGL2/3 homolog
* SPAPB1A10.11c mitochondrial glutamyl-tRNA ligase Mse1
* SPCC1183.04c mitochondrial RNA metabolism pathway protein Pet127
* SPAC22F3.06c Lon1
* SPBC1347.07 RNA exonuclease Rex2, processes RNA in nuc and mito, multiple differentially localized paralogs in Scer.
* SPBC3F6.04c U3 snoRNP protein Nop14 
* SPBC16E9.06c BolA domain UV induced protein Uvi31
* SPBC146.12 monooxygenase Coq6 
* SPBC3D6.03c mitochondrial 3'-tRNA processing endonuclease tRNAse Z, Trz2, mito ortholog of Trz1, and Trz1 is dual-localized in Scer.
* SPBC16A3.14 superoxide dismutase, mitochondrial ribosomal protein subunit
* SPBC543.09 mitochondrial m-AAA protease Yta12
* SPBC2G2.12 mitochondrial and cytoplasmic histidine-tRNA ligase Hrs1
* SPBC1539.01c mitochondrial ribosomal protein subunit L15 Mrp15
* SPAC4A8.03c protein phosphatase 2C Ptc4
* SPBC1677.03c threonine ammonia-lyase Tda1. Scer paralogs ILV1 (mito) and SRY1 (loc not known)
* SPBC146.13c myosin type I
* SPAC1B3.04c mitochondrial elongation factor GTPase Guf1 
* SPAC6C3.04 citrate synthase Cit1. Scer has two paralogs, Cit1 (mito) and Cit2 (cyto/peroxisomal). But here the 2nd ATG is right next to the first ATG so that's probably just an annotaion off by 1.
* SPBC15C4.02 ABC1 kinase family protein, implicated in mito activity, Mcp2 homolog

Histidine tRNA Ligase, Hurrah!

Several mitochondrial ribosomal proteins, check these.

Mostly the 2nd ATG is not in frame though. Restricting to frame gives only Rex2 and HisRS. 

### less strong biases, but in frame dATG

```{r highdiffn_dvsaATG_mitopred_inframe,dependson="dvsaATG_scoresn",results="show"}
scores %>%
    arrange(desc(d1vsan)) %>%
    select(Gene,aATG.scorekn,d1.scorekn) %>%
    inner_join( ctbl_big %>% 
                   filter(d1.frame==0) %>% 
                   select(Gene,d1.posATG),
               by="Gene" ) %>%
    left_join( ribosum ) %>%
    filter(Gene %in% mitopredlist) %>%
    print(n=20)
```

* SPBC2G2.08 ade9 C-1-tetrahydrofolatesynthase &c, Scer has MIS1 (mito) & ADE3 (cyto) homologs. Predicted cleavage site is 12aas in.
* SPBC21C3.04c mitochondrial ribosomal protein subunit L34, but ATGATG
* SPAC3H1.10 phytochelatin synthetase
* SPAC4F8.02c mitochondrial ribosomal protein subunit L40
* SPAC23C11.17 mdm28 mitochondrial inner membrane protein potassium transport
* SPAC26F1.14c apoptosis-inducing factor homolog Aif1, orthologs dual-localized? No Scer homolog.
* SPBC1861.05 pseudouridine-metabolizing bifunctional protein.
* SPBC2G2.04c mitochondrial matrix protein, YjgF family protein Mmf1, Scer has  mito (MMF1) and cyto (HMF1) paralogs.
* SPCC1682.01 qcr9 ubiquinol-cytochrome-c reductase complex subunit 9

Some are dual-localized.

Predict that *many* more of the dual-localized things, such as aa-tRNA-synthetases, have non-ATG starts.


<a href="#top">Back to table of contents</a>
