---
title: "Start codon context for Schizosaccharomyces pombe"
author: Edward Wallace, ewjwallace@gmail.com
date: 6 Oct 2018
output:
  # html_notebook: default
  html_document:
    toc: true
    toc_depth: 4
---

# Summary

This presents the analysis of Schizosaccharomyces start codon usage and context. This uses the S. pombe 972h- translation estimates from Duncan and Mata 2017.

### Load Packages

```{r setup,message=FALSE,echo=FALSE,cache=FALSE}
knitr::opts_chunk$set(warning=FALSE,message=FALSE,echo=FALSE,cache=TRUE,
                      results="hide",
                      fig.path="figure/SpombeATGContext-",
                      cache.path="cache/SpombeATGContext-")
library(tidyverse)
library(cowplot)
library(ggrepel)
library(ggseqlogo)
library(Biostrings)

theme_set(theme_cowplot(font_size=12) + 
              theme(strip.background = element_blank()) )



geom_diagline <- function(linetype='solid',size=0.1,colour="grey20",...) {
    geom_abline(slope=1,intercept=0,linetype=linetype,colour=colour)
}

scientific_10 <- function(x) {
    xout <- gsub("1e", "10^{", format(x),fixed=TRUE)
    xout <- gsub("{-0", "{-", xout,fixed=TRUE)
    xout <- gsub("{+", "{", xout,fixed=TRUE)
    xout <- gsub("{0", "{", xout,fixed=TRUE)
    xout <- paste(xout,"}",sep="")
    return(parse(text=xout))
}

scale_x_log10nice <- function(name=waiver(),omag=seq(-10,20),...) {
    breaks10 <- 10^omag
    scale_x_log10(name,breaks=breaks10,labels=scientific_10(breaks10),...)
}

scale_y_log10nice <- function(name=waiver(),omag=seq(-10,20),...) {
    breaks10 <- 10^omag
    scale_y_log10(name,breaks=breaks10,labels=scientific_10(breaks10),...)
}

scale_loglog <- function(...) {
    list(scale_x_log10nice(...),scale_y_log10nice(...))
}

round3 <- function(x) round(x,digits=3)

logmean <- function(x) {
    lx <- log(x)
    exp( mean( lx[is.finite(lx)] ) )
}
# setwd("~/Repos/CryptoTranscriptome2018/ATGContextOthers/")
```


# Here we go

## Expression: RNA abundance and ribosome-protected-fragments

### Load expression data

```{r load_ribosum_Sp,results="show"}

tpms_Sp <- read_tsv("data_in/ExpressionTranslation/SpombeDuncan2017_TPMs_collated.tsv") 

ribosum_Sp <- tpms_Sp %>%
    mutate(RPF=(NCHX1+NCHX3+NnoC1+NnoC3)/4,
           RPF_noN=(SCHX1+SCHX3+SnoC1+SnoC3)/4) %>%
    select(Gene=ORF,RPF,RPF_noN)
    
ribosum_Sp %>%
    arrange(desc(RPF)) 

hiTrans_Sp <- ribosum_Sp %>%
    arrange(desc(RPF)) %>%
    head(n=256) %>%
    .$Gene

midTrans_Sp <- ribosum_Sp %>%
    arrange(desc(RPF)) %>%
    head(n=1280) %>%
    .$Gene

# enoughRNA_Sp <- ribosum_Sp %>%
#     arrange(desc(RNA)) %>%
#     head(n=3000) %>%
#     .$Gene
```

#### Check replicability

```{r scatter_ribo_Sp,dependson="load_ribosum_Sp",results="show",fig.width=7,fig.height=7}
library(GGally)

ggpairs(data=tpms_Sp %>% select(NCHX1:NnoC3) %>% log10())
```

### Ribosome occupancy mostly tracks RNA abundance

```{r plot_TE_Sp,dependson="load_ribosum_Sp",fig.height=3.9,fig.width=4.1,eval=FALSE}
TE_unusual_Sp <- 
    ribosum_Sp %>%
    filter( (RPF > 1.8e4) |
                (RPF > 100 & RPF > RNA * 5 ) | 
                (RNA > 150 & RPF < RNA / 10 ))


ggplot(data=ribosum_Sp,aes(x=RNA,y=RPF)) +
    geom_diagline() +
    geom_point(size=.8,colour="grey50") +
    geom_point(data=TE_unusual_Sp,aes(colour=Gene),size=1) + 
    geom_text_repel(data=TE_unusual_Sp,aes(label=Gene,colour=Gene)) +
    scale_loglog(limits=c(1e-1,2e4),
                  oob=scales::squish,
                  expand=c(0.01,0.01)) +
    labs(x="RNA abundance (TPM of aORF)",
         y="Ribosome occupancy (TPM of aORF)") +
    theme(legend.position="none")
```

## ATG Context

### Load context data

```{r load_context_Sp,results="show"}
ctbl_big_Sp <- read_tsv("data_in/SPombe_allATGcontext.table.txt",
                        skip=1,
                        col_names=c("gene","aATG.context","aATG.pos",
                                    "d1.context","d1.posTSS","d1.posATG","d1.frame")
                        ) %>% 
    dplyr::rename(Gene=gene) %>%
    filter(!is.na(aATG.context) ) %>%
    mutate(aATG.context=toupper(aATG.context))
ctbl_big_Sp
```

### Annotated ATGs have a Kozak consensus sequence

```{r aATG_context_Sp,fig.width=4,fig.height=1}
ggseqlogo(data=ctbl_big_Sp$aATG.context,namespace="TCAG") + 
    theme_nothing()
```

### Highly translated Annotated ATGs have a Kozak consensus sequence

That's for hiTrans_Sp, the top 5% (256) translated genes by RPF TPM.

```{r a_hi_context_Sp,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big_Sp,Gene %in% hiTrans_Sp)$aATG.context,
                      namespace="TCAG") + 
    theme_nothing()
```



### Downstream ATGs don't have a consensus

First downstream ATG

```{r d1_context_Sp,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big_Sp,nchar(d1.context) ==24)$d1.context,
                      namespace="TCAG") + 
    theme_nothing()
```

### Downstream ATGs in frame and highly translated don't have a consensus

Except for 3rd-codon-position bias.

```{r d1_hi_frame0_context_Sp,fig.width=4,fig.height=1}
ggseqlogo(data=filter(ctbl_big_Sp,
                      nchar(d1.context) == 24,
                      d1.frame==0,
                      Gene %in% hiTrans_Sp)$d1.context,
                      namespace="TCAG") + 
    theme_nothing()
```

## Calculate Information content and scores of consensus motif

### Calculate a wide and a narrow consensus sequence

Calculate motif score against the position weight matrix (pwm) for both **narrow** (-5 from ATG through to ATG) and **wide** (-9 from ATG to +3) kozak consensus motif. These motifs are taken from the top 5% highly translated genes.

```{r kozak_consensus_Sp,fig.width=4,fig.height=1}
kozak_w_PFM_Sp <- filter(ctbl_big_Sp,Gene %in% hiTrans_Sp) %>%
    .$aATG.context %>%
    str_sub(start=4L,end=18L) %>%
    str_to_upper() %>%
    consensusMatrix() 

kozak_w_PWM_Sp <- PWM(kozak_w_PFM_Sp)
kozak_n_PWM_Sp <- kozak_w_PFM_Sp[,5:12] %>% PWM()
```


```{r write_PFM_Sp, dependson="calc_Spores_Sp",results="show"}
cat(
    "# PFM_hiTrans_Sp.txt
# Position Frequency Matrix for aATGs of 256 highest-translated genes from Sc
# This is of the wide ATG context NNNNNNNNNATGNNN
# Column is position, Row is nt count in order: ACGT.
#
# Edward Wallace, June 2018
",
file="data_out/PFM_hiTrans_Sp.txt")
kozak_w_PFM_Sp %>%
    as.tibble() %>%
    write_tsv("data_out/PFM_hiTrans_Sp.txt",
              append = TRUE,col_names=FALSE)
```

### Estimate the information content 

Using the sequence logo, details on https://en.wikipedia.org/wiki/Sequence_logo

```{r kozak_inf_Sp,fig.width=4,fig.height=1}
entropy1 <- function(counts,ssize=4,pseudocount=0.5) {
    counts <- counts + pseudocount
    ntot = sum(counts)
    freqs = counts / sum(counts)
    Hentropy = - sum( freqs * log2(freqs) ) 
    ecorrection = ( ssize - 1 ) / ( log(2) *  ntot )
    information = log2(ssize) - Hentropy
    return(information)
}

entropytot <- function(pfm,ssize=4,pseudocount=0.5) {
    apply(pfm,2,entropy1) %>% sum()
}

entropytot(kozak_w_PFM_Sp)
# entropytot(kozak_w_PFM_Sp[,5:12])
entropytot(kozak_w_PFM_Sp[,-(10:12)])
entropytot(kozak_w_PFM_Sp[,5:9])
```

Information content in bits of highly-translated consensus (excluding 6 bits from ATG), **narrow** is `r entropytot(kozak_w_PFM_Sp[,5:9]) %>% round(2)`, of **wide** is `r entropytot(kozak_w_PFM_Sp[,-(10:12)]) %>% round(2)`.

### Calculate scores of aATG, dATG, uATG against Kozak consensus

```{r calc_Spores_Sp, dependson="kozak_consensus_Sp"}

PWMscore <- function(seqs,pwm,startl) {
    # calculate scores for a character vector seqs against
    # pwm from given starting position
    # after checking strings are non-missing and start with ACTG
    vapply(str_sub(seqs,start=startl),
           function(ss) {
               if ( is.na(ss) ) {
                   return(NA)
               } else if ( str_detect(ss,"\\A[ACTG]") ) {
                   return( PWMscoreStartingAt(pwm,ss) )
               } else {
                   return(NA) 
               } 
           } ,
           FUN.VALUE = 0)
}

PWMscoren <- function(seqs,pwm=kozak_n_PWM_Sp,startl=8) {
    PWMscore(seqs,pwm,startl) 
}

PWMscorew <- function(seqs,pwm=kozak_w_PWM_Sp,startl=4) {
    PWMscore(seqs,pwm,startl) 
}

is.context <- function(x) str_detect(x,"context")

scores_Sp <- 
    ctbl_big_Sp %>%
    transmute(Gene,
              aATG.scorekn = PWMscoren(str_to_upper(aATG.context)),
              d1.scorekn   = PWMscoren(d1.context),
              # u1.scorekn   = PWMscoren(u1.context),
              # aATG.scorekw = PWMscorew(str_to_upper(aATG.context)),
              # d1.scorekw   = PWMscorew(d1.context),
              # u1.scorekw   = PWMscorew(u1.context),
              NULL
    ) %>%
    mutate(d1vsan = d1.scorekn - aATG.scorekn,
           # u1vsan = u1.scorekn - aATG.scorekn,
           # d1vsaw = d1.scorekw - aATG.scorekw,
           # u1vsaw = u1.scorekw - aATG.scorekw,
           NULL) 

# ribotbl_Sporesenough_Sp <-
#     ribosum_Sp %>%
#     filter(Gene %in% enoughRNA_Sp) %>%
#     # left_join(ribotbl_Sp,by="Gene") %>% 
#     left_join(scores_Sp,by="Gene")

```

We calculate scores using Biostrings::PWMscoreStartingAt.

The best description I could find of this method is: https://support.bioconductor.org/p/61520/

It is just the sum of the matrix product of the PWM with the sequence.


Write scores to file *scores_kozak_Sp.txt*.

```{r write_Spores_Sp, dependson="calc_Spores_Sp",results="show"}
scores_Sp
cat(
    "# scores_kozak_Sp.txt
# Scores of ATG context against Kozak sequence, C. neoformans serotype D Sc
# Kozak sequence derived from 256 highest-translated genes by RPF
#
# Gene: Sc systematic gene name
# aATG.scorekn: score of annotated ATG against narrow Kozak sequence (NNNNNATG)
# d1.scorekn: score of 1st downstream ATG against narrow
# xx.scorekw: analogous scores against wide Kozak sequence (NNNNNNNNNATGNNN)
#
# Edward Wallace, June 2018
",
file="data_out/scores_kozak_Sp.txt")
scores_Sp %>%
    mutate_if(is.numeric,round3) %>%
    write_tsv("data_out/scores_kozak_Sp.txt",
              append = TRUE,col_names=TRUE)
```

### Plot against narrow consensus (-5 to ATG)

```{r plot_Spores_n_Sp, dependson="calc_Spores_Sp", fig.width=5,fig.height=1.8}

sbandwidthn = 0.025

ggplot(data=scores_Sp) +
    geom_density(aes(x=aATG.scorekn,colour="aATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    geom_density(aes(x=d1.scorekn,colour="dATG"),
                 kernel="rectangular",bw=sbandwidthn) +
    # geom_density(aes(x=u1.scorekn,colour="uATG"),
    #             kernel="rectangular",bw=sbandwidthn) +
    geom_density(data=filter(scores_Sp,Gene %in% hiTrans_Sp),
                 aes(x=aATG.scorekn,colour="aATG, top 5%"),
                 kernel="rectangular",bw=sbandwidthn) +
    scale_colour_manual(values=c("aATG"="dodgerblue",
                                 "aATG, top 5%"="darkblue",
                                 # "uATG"="red",
                                 "dATG"="green3") ) +
    labs(x="ATG Kozak score, Sc") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank(),
          legend.title=element_blank())
```

## Compare aATG and dATG context by gene

### Most dATG scores are less than aATG scores

```{r dvsaATG_Sporesdens_Sp, dependson="calc_Spores_Sp", fig.width=3.5,fig.height=1.8}

ggplot(data=scores_Sp) +
    geom_vline(xintercept=0,size=0.3) +
    geom_density(aes(x=d1vsan,colour="narrow"),kernel="rectangular") +
    # geom_density(aes(x=d1vsaw,colour="wide"),kernel="rectangular")  +
    labs(x="difference in score, dATG vs aATG",
         colour="context") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank(),
          legend.position=c(0.8,0.8))
```

### For highly translated genes, most dATG scores are much less than aATG

```{r dvsaATG_Sporesn_Sp, dependson="calc_Spores_Sp", fig.width=3.5,fig.height=3.5}
highdiffn_dvsaATG_Sp <- scores_Sp %>%
    arrange(desc(d1vsan)) %>%
    head(n=256) %>%
    select(Gene,aATG.scorekn,d1.scorekn)

ggplot(data=scores_Sp %>% filter(!is.na(d1.scorekn)),
       aes(x=aATG.scorekn,y=d1.scorekn)) +
    geom_diagline() +
    geom_point(size=0.5,colour="grey50") +
    geom_point(data=highdiffn_dvsaATG_Sp,
               colour="red",size=1) +
    geom_point(data=scores_Sp %>% 
                   filter(Gene %in% hiTrans_Sp),
               colour="blue",size=1) +
    geom_point(data=highdiffn_dvsaATG_Sp %>% 
                   filter(Gene %in% hiTrans_Sp),
               colour="purple4",size=1.3) +
    labs(x="aATG score, narrow",
         y="dATG score, narrow")
```

Red: high dATG vs aATG **Kozak** score.
Blue: highly translated.

(Purple: both. None of the red genes are highly translated.)

### Small negative correlation between dATG and aATG score

R = `r with(scores_Sp, cor(aATG.scorekn,d1.scorekn,use="pairwise.complete.obs")) %>% round3`

## Genes with unusual dATG vs aATG narrow score

Those genes are in this list:

```{r highdiffn_dvsaATG_Sp,dependson="dvsaATG_Sporesn_Sp",results="show"}
highdiffn_dvsaATG_Sp
```

* SPAC222.04c/Ies6, Ino80 chromatin remodeling complex subunit
* SPAC26F1.05/Mug106, meiotically upregulate, short, no known orthologs.
* SPCC1620.12c GTPase activating protein, distand MDR1/GYP2 homolog
* SPBC9B6.11c CCR4/nocturin family endoribonuclease, NGL2/3 homolog
* SPAC12B10.01c HECT-type ubiquitin-protein ligase E3, UFD4 homolog
* SPAC1B2.06, short, no known orthologs


Some of these are decently translated (top 25%)

```{r highdiffn_dvsaATG_hiTrans_Sp,dependson="dvsaATG_Sporesn_Sp",results="show"}
highdiffn_dvsaATG_Sp %>% 
                   filter(Gene %in% midTrans_Sp) 
```

* SPCC31H12.08c CCR4-Not complex subunit Ccr4
* SPBC13G1.01c mitochondrial ribosomal protein subunit S4
* SPCC188.04c NMS complex subunit Spc25
* SPBC1198.08 dipeptidase Dug1
* SPAC1071.01c mRNA cleavage and polyadenylation specificity factor complex subunit Pta1
* SPAC22F3.06c Lon protease homolog Lon1 (mito localized)
* SPBC428.01c nucleoporin Nup107
* SPBC3F6.04c U3 snoRNP protein Nop14

Several things involved in mRNA 3' end regulation. Otheriwse unclear. We should check if those ATGs are actually used.

## Compare score difference to localization predictions


### Load predictions from mitofates

In input file *SPombe_mitofates.txt*.

```{r load_mitofates_Sp}
mitofates_Sp <- read_tsv("data_in/SPombe_mitofates.txt",
                            comment="#",skip=1,
                         col_names=c("Gene","Prob_preseq","Pred_preseq","CSite_enzyme",
"Net_charge","Pos_TOM20","Pos_afhelix",
"BHHPPP","BPHBHH","HBHHBb","HBHHbB","HHBHHB","HHBPHB","HHBPHH",
"HHBPHP","HHHBBH","HHHBPH","HHHHBB","HHPBHH","HPBHHP","PHHBPH")) %>%
    mutate(Pred_preseq=factor(Pred_preseq,
                              levels=c("No mitochondrial presequence",
                                       "Possessing mitochondrial presequence"),
                              labels=c("No","Yes")),
           Gene = str_remove(Gene, " Protein"))
mitocounts_Sp <- 
    mitofates_Sp %>% 
    group_by(Pred_preseq) %>% 
    count() 
mitobreaks_Sp <- mitocounts_Sp %>%
    mutate(label=paste0(Pred_preseq," (n = ",n,")")) %>%
    .$label %>%
    set_names(mitocounts_Sp$Pred_preseq)

mitopredlist_Sp <- mitofates_Sp %>%
    filter(Pred_preseq =="Yes") %>%
    .$Gene
```


### Sp Genes with high dATG vs aATG score *are* enriched in mitochondrial presequences

```{r dvsaATG_Sporesdiffn_mitofates_Sp, dependson=c("calc_Spores_Sp","load_mitofates_Sp"), fig.width=5,fig.height=1.8}

ggplot(data=scores_Sp %>% 
           filter(!is.na(d1.scorekn)) %>%
           left_join( mitofates_Sp %>% 
                          select(Gene,Prob_preseq,Pred_preseq)
                      ) ,
       aes(x=d1.scorekn-aATG.scorekn,colour=Pred_preseq)) +
    geom_vline(xintercept=0,size=0.3) +
    geom_density(kernel="rectangular",bw=0.025) +
    scale_colour_manual(values=c("Yes"="magenta","No"="grey50"),
                        labels=mitobreaks_Sp)  + 
    labs(x="Kozak score, difference dATG - aATG", colour="Mito. pred") +
    theme(axis.line.y=element_blank(),axis.ticks.y=element_blank(),
          axis.text.y=element_blank(),axis.title.y=element_blank())
```

### Sp mito-localized genes also do not have a distinctive aATG context

Although the +5 T is striking.

```{r aATG_motif_mitofates_Sp, dependson=c("calc_Spores_Sp","load_mitofates_Sp"), fig.width=4,fig.height=1}

ggseqlogo(data=ctbl_big_Sp %>%
              filter(Gene %in% mitopredlist_Sp) %>%
              .$aATG.context,
          namespace="TCAG")  + 
    theme_nothing()

```

## Genes with unusual dATG vs aATG narrow score AND predicted mito loc

Those genes are in this list:

```{r highdiffn_dvsaATG_mitopred_Sp,dependson="dvsaATG_Sporesn_Sp",results="show"}
highdiffn_dvsaATG_Sp %>%
    left_join( ctbl_big_Sp %>% select(Gene, d1.frame,d1.posATG) ) %>%
    left_join( ribosum_Sp ) %>%
    filter(Gene %in% mitopredlist_Sp) %>%
    print(n=26)
```

* SPBC13G1.01c mitochondrial ribosomal protein subunit S4
* SPBC9B6.11c CCR4/nocturin family endoribonuclease, NGL2/3 homolog
* SPAPB1A10.11c mitochondrial glutamyl-tRNA ligase Mse1
* SPCC1183.04c mitochondrial RNA metabolism pathway protein Pet127
* SPAC22F3.06c Lon1
* SPBC1347.07 RNA exonuclease Rex2, processes RNA in nuc and mito, multiple differentially localized paralogs in Scer.
* SPBC3F6.04c U3 snoRNP protein Nop14 
* SPBC16E9.06c BolA domain UV induced protein Uvi31
* SPBC146.12 monooxygenase Coq6 
* SPBC3D6.03c mitochondrial 3'-tRNA processing endonuclease tRNAse Z, Trz2, mito ortholog of Trz1, and Trz1 is dual-localized in Scer.
* SPBC16A3.14 superoxide dismutase, mitochondrial ribosomal protein subunit
* SPBC543.09 mitochondrial m-AAA protease Yta12
* SPBC2G2.12 mitochondrial and cytoplasmic histidine-tRNA ligase Hrs1
* SPBC1539.01c mitochondrial ribosomal protein subunit L15 Mrp15
* SPAC4A8.03c protein phosphatase 2C Ptc4
* SPBC1677.03c threonine ammonia-lyase Tda1. Scer paralogs ILV1 (mito) and SRY1 (loc not known)
* SPBC146.13c myosin type I
* SPAC1B3.04c mitochondrial elongation factor GTPase Guf1 
* SPAC6C3.04 citrate synthase Cit1. Scer has two paralogs, Cit1 (mito) and Cit2 (cyto/peroxisomal). But here the 2nd ATG is right next to the first ATG so that's probably just an annotaion off by 1.
* SPBC15C4.02 ABC1 kinase family protein, implicated in mito activity, Mcp2 homolog

Histidine tRNA Ligase, Hurrah!

Several mitochondrial ribosomal proteins, check these.

Mostly the 2nd ATG is not in frame though. Restricting to frame gives only Rex2 and HisRS. 

### less strong biases, but in frame dATG

```{r highdiffn_dvsaATG_mitopred_inframe_Sp,dependson="dvsaATG_Sporesn_Sp",results="show"}
scores_Sp %>%
    arrange(desc(d1vsan)) %>%
    select(Gene,aATG.scorekn,d1.scorekn) %>%
    inner_join( ctbl_big_Sp %>% 
                   filter(d1.frame==0) %>% 
                   select(Gene,d1.posATG),
               by="Gene" ) %>%
    left_join( ribosum_Sp ) %>%
    filter(Gene %in% mitopredlist_Sp) %>%
    print(n=20)
```

* SPBC2G2.08 ade9 C-1-tetrahydrofolatesynthase &c, Scer has MIS1 (mito) & ADE3 (cyto) homologs. Predicted cleavage site is 12aas in.
* SPBC21C3.04c mitochondrial ribosomal protein subunit L34, but ATGATG
* SPAC3H1.10 phytochelatin synthetase
* SPAC4F8.02c mitochondrial ribosomal protein subunit L40
* SPAC23C11.17 mdm28 mitochondrial inner membrane protein potassium transport
* SPAC26F1.14c apoptosis-inducing factor homolog Aif1, orthologs dual-localized? No Scer homolog.
* SPBC1861.05 pseudouridine-metabolizing bifunctional protein.
* SPBC2G2.04c mitochondrial matrix protein, YjgF family protein Mmf1, Scer has  mito (MMF1) and cyto (HMF1) paralogs.
* SPCC1682.01 qcr9 ubiquinol-cytochrome-c reductase complex subunit 9

Some are dual-localized.

Predict that *many* more of the dual-localized things, such as aa-tRNA-synthetases, have non-ATG starts.


<a href="#top">Back to table of contents</a>
