---
title: "Near-AUG starts for Fungi"
author: Edward Wallace, ewjwallace@gmail.com
date: 12 Oct 2018
output:
  # html_notebook: default
  html_document:
    toc: true
    toc_depth: 4
---

# Summary


Near-AUG near-cognate start codons.

Algorithm is described in function `estimate_3prime_nrstart_CDS`.

- take 120nt upstream from start, from files `*_CDS_with_120bputrs.fa`
- trim 5' end at last in-frame stop codon to remove uORFs, with `trim_at_last_stop_codon`
- trim 5' end to first near-AUG start, with `trim_to_first_nrstart`
- output entire N-terminally extended CDS to file `*_protein_with_nrAUG_Ntermext.fa`

We used as near-starts: ATG, TTG, CTG, ACG, ATT, GTG, ATA.

Note we automatically translate near-ATGs according to the standard genetic code, not necessarily to `M`.


### Load Packages

```{r setup,message=FALSE,echo=FALSE,cache=FALSE}
knitr::opts_chunk$set(warning=FALSE,message=FALSE,echo=TRUE,cache=FALSE,
                      results="show",
                      fig.path="figure/nearAUGstarts-",
                      cache.path="cache/nearAUGstarts-")
# load common functions
source("../SharedScripts/sharedScripts.R")
# setwd("~/Repos/CryptoTranscriptome2018/ATGContext/")
```

## Process H99

```{r process_H99,results="show"}
expand_threeprime_nrstarts(txinfile = "../CDSFasta/data_in/H99_10p_up12dwn9_CDS_with_120bputrs.fa",
                        CDSoutfile="../CDSFasta/data_out/H99_10p_up12dwn9_CDS_with_nrAUG_5pext.fa",
                        protoutfile="../ProteinFasta/data_out/H99_10p_up12dwn9_protein_with_nrAUG_Ntermext.fa")
```

## Process JEC21

```{r process_JEC21,results="show"}
expand_threeprime_nrstarts(txinfile = "../CDSFasta/data_in/JEC21_10p_up12dwn9_CDS_with_120bputrs.fa",
                        CDSoutfile="../CDSFasta/data_out/JEC21_10p_up12dwn9_CDS_with_nrAUG_5pext.fa",
                        protoutfile="../ProteinFasta/data_out/JEC21_10p_up12dwn9_protein_with_nrAUG_Ntermext.fa")
```

## Process Ca

```{r process_Ca,results="show"}
expand_threeprime_nrstarts(txinfile = "../CDSFastaOthers/data_in/CAlbi_CDS_with_120bputrs.fa",
                        CDSoutfile="../CDSFastaOthers/data_out/CAlbi_CDS_with_nrAUG_5pext.fa",
                        protoutfile="../ProteinFastaOthers/data_out/CAlbi_protein_with_nrAUG_Ntermext.fa")
```

## Process Hc - not done

Histoplasma capsulatum data not uploaded.

```{r process_Hc,results="show",eval=FALSE}
expand_threeprime_nrstarts(txinfile = "../CDSFastaOthers/data_in/Histo_CDS_with_120bputrs.fa",
                        CDSoutfile="../CDSFastaOthers/data_out/Histo_CDS_with_nrAUG_5pext.fa",
                        protoutfile="../ProteinFastaOthers/data_out/Histo_protein_with_nrAUG_Ntermext.fa")
```

## Process Nc

```{r process_Nc,results="show"}
expand_threeprime_nrstarts(txinfile = "../CDSFastaOthers/data_in/NC12_CDS_with_120bputrs.fa",
                        CDSoutfile="../CDSFastaOthers/data_out/NC12_CDS_with_nrAUG_5pext.fa",
                        protoutfile="../ProteinFastaOthers/data_out/NC12_protein_with_nrAUG_Ntermext.fa")
```

## Process Sc

```{r process_Sc,results="show"}
expand_threeprime_nrstarts(txinfile = "../CDSFastaOthers/data_in/SacCer_CDS_with_120bputrs.fa",
                        CDSoutfile="../CDSFastaOthers/data_out/SacCer_CDS_with_nrAUG_5pext.fa",
                        protoutfile="../ProteinFastaOthers/data_out/SacCer_protein_with_nrAUG_Ntermext.fa")
```

## Process Sp

```{r process_Sp,results="show"}
expand_threeprime_nrstarts(txinfile = "../CDSFastaOthers/data_in/SPombe_CDS_with_120bputrs.fa",
                        CDSoutfile="../CDSFastaOthers/data_out/SPombe_CDS_with_nrAUG_5pext.fa",
                        protoutfile="../ProteinFastaOthers/data_out/SPombe_protein_with_nrAUG_Ntermext.fa")
```



