---
title: "Downstream ATG depletion and other position-specific codon frequencies for Cryptococcus neoformans"
author: Edward Wallace, ewjwallace@gmail.com
date: 5 Nov 2018
output:
  # html_notebook: default
  html_document:
    toc: true
    toc_depth: 4
---

# Summary

This assesses the extent to which coding sequences are depleted in ATGs. It extends to a more general analysis of codon frequencies dependent on position and frame, metagene-style. This uses the best-transcript annotation and corresponding start codon position and sequence map made by Corinne Maufrais in June 2018.


### Load Packages

```{r setup,message=FALSE,echo=FALSE,cache=FALSE}
knitr::opts_chunk$set(warning=FALSE,message=FALSE,echo=FALSE,cache=TRUE,
                      results="hide",
                      fig.path="figure/CryptodATGdepletion-",
                      cache.path="cache/CryptodATGdepletion-")
# load common functions
source("../SharedScripts/sharedScripts.R")

# setwd("~/Repos/CryptoTranscriptome2018/ATGContext/")
```


## Load CDS

```{r load_cds}
CDS <- readDNAStringSet("../CDSFasta/data_in/JEC21_10p_up12dwn9_CDS.fa")

countCodonsAll <- function(sset) {
    sset %>%
        as.character %>%
        lapply(str_dice,width=3L) %>%
        unlist %>%
        table
}

# debugonce(countCodonsAll)
CDS %>% head %>% countCodonsAll

tidyPos <- function(x,variable="x") {
    tibble(1:length(x),x) %>%
        set_names(c("Pos",variable))
}

groupThingPos <- function(sset,width=3L,variable="Codon") {
    sset %>%
        as.character %>%
        lapply(str_dice,width=width) %>%
        lapply(tidyPos,variable=variable) %>%
        bind_rows %>%
        group_by(Pos)
}

countCodonsPos <- function(sset,width=3L,variable="Codon") {
    groupThingPos(sset,width=width,variable=variable) %>%
        count(Codon) %>%
        mutate(Codon=factor(Codon,levels=names(GENETIC_CODE)),
            Freq=n/sum(n))
}

ccpos <- 
    CDS %>% 
    countCodonsPos %>%
    ungroup

# ggplot(data=ccpos %>% filter(Codon=="ATG"),aes(x=Pos,y=Freq)) + 
#     geom_line() + 
#     scale_x_continuous(limits=c(2,200)) + 
#     expand_limits(y=0) + scale_y_continuous(limits=c(0,0.1),expand=c(0,0))

sensecodons <- names(GENETIC_CODE) %>%
    setdiff(c("TGA","TAA","TAG"))
```

### ATG Codons in-frame are depleted near start, but others are more depleted.

This is smoothed to make it readable. 

```{r plot_ATG_depletion,dependson="load_cds",fig.height=3,fig.width=5}
ggplot(data=ccpos %>% filter(Codon %in% sensecodons),
       aes(x=Pos,y=Freq)) + 
    geom_smooth(aes(group=Codon),colour="grey50",
                size=0.4,se=FALSE,span=0.1,method="loess") + 
    geom_smooth(data=ccpos %>% filter(Codon == "ATG"),
              colour="red",size=1.2,se=FALSE,span=0.1,method="loess") + 
    scale_x_continuous("Position",limits=c(2,400),expand=c(0,1)) + 
    expand_limits(y=0) + 
    scale_y_continuous("Codon Frequency In-Frame",expand=c(0,0))

```

```{r plot_ATGplus_depletion,dependson="load_cds",fig.height=3,fig.width=6}
ggplot(data=ccpos %>% filter(Codon %in% sensecodons),
       aes(x=Pos,y=Freq)) + 
    geom_smooth(aes(group=Codon),colour="grey50",
                size=0.4,se=FALSE,span=0.1,method="loess") + 
    geom_smooth(data=ccpos %>% filter(Codon == "CCC"),
                aes(colour="CCC"),size=0.9,se=FALSE,span=0.1,method="loess") + 
    geom_smooth(data=ccpos %>% filter(Codon == "TGT"),
                aes(colour="TGT"),size=0.9,se=FALSE,span=0.1,method="loess") + 
    geom_smooth(data=ccpos %>% filter(Codon == "GAA"),
                aes(colour="GAA"),size=0.9,se=FALSE,span=0.1,method="loess") +
    geom_smooth(data=ccpos %>% filter(Codon == "ATG"),
                aes(colour="ATG"),size=1.2,se=FALSE,span=0.1,method="loess") + 
    scale_x_continuous("Position",limits=c(2,400),expand=c(0,1)) + 
    expand_limits(y=0) + 
    scale_y_continuous("Codon Frequency In-Frame",expand=c(0,0)) +
    scale_colour_manual("Selected\nCodons",values=c(ATG="red",CCC="forestgreen",
                                         TGT="purple1",GAA="purple4"))

```

### Other in-frame codons have very different depletion/enrichment

```{r plot_ccposframe0,dependson="load_cds",fig.width=7,fig.height=10}
ggplot(data=ccpos %>% filter(Codon %in% names(GENETIC_CODE)),
       aes(x=Pos,y=Freq)) +
    # geom_line(aes(group=Frame)) +
    geom_smooth(colour="green4",
                se=FALSE,span=0.1,method="loess") +
    scale_x_continuous("Position",limits=c(2,400),expand=c(0,1)) + 
    expand_limits(y=0) + 
    scale_y_continuous("Codon Frequency In-Frame",expand=c(0,0),
                       breaks=c(0,0.02,0.04)) +
    # scale_colour_brewer(palette="Set2") +
    facet_wrap(~Codon,ncol=4) +
    theme(legend.position="top", 
          panel.border = element_rect(colour = "grey80",size = 0.5,
                                      fill=NA,linetype=1) )
```

### What are the non-ATG codons depleted near start?

GAA, GAT, GAG. Why?

```{r altdepletion,dependson="load_cds"}
ggplot(data=ccpos %>% 
           filter(Codon %in% sensecodons,
                  Pos %in% c(2,3,10,100)) %>%
           mutate(Pos = factor(Pos)),
       aes(x=Pos,y=Freq,colour=Codon,label=Codon)) + 
    geom_text() + 
    geom_line(aes(group=Codon),alpha=0.5) + 
    scale_y_continuous(limits=c(0,0.1),expand=c(0,0)) +
    theme(legend.position="none")
```

### GAA, GAT, GAG, TGT, are more depleted near-start than ATG, and CTC is strange

But not GAC. Why both E codons only one D codon?

```{r depletionextreme,dependson="load_cds"}
ggplot(data=ccpos %>% 
           filter(#Codon %in% sensecodons,
               Codon %in% c("GAA","GAT","GAG","GAC","ATG","CTC","TGT","GGT"),
                  # str_sub(Codon,end=2L) == "GA",
                  Pos %in% c(2,3,10,100)) %>%
           mutate(Pos = factor(Pos)),
       aes(x=Pos,y=Freq,colour=Codon,label=Codon)) + 
    geom_text() + 
    geom_line(aes(group=Codon)) + 
    scale_y_continuous(limits=c(0,0.05),expand=c(0,0)) +
    theme(legend.position="none")
```

### TCN, GCC, CCC, are enriched near the start

Mostly serine. Is that the N-end rule for degradation at work or a translation initiation phenomenon?

```{r enrichmentextreme,dependson="load_cds"}
ggplot(data=ccpos %>% 
           filter(#Codon %in% sensecodons,
               Codon %in% c("TCT","TCC","TCA","TCG",
                            "GCC","CCC","ATG"),
                  # str_sub(Codon,end=2L) == "GA",
                  Pos %in% c(2,3,10,100)) %>%
           mutate(Pos = factor(Pos)),
       aes(x=Pos,y=Freq,colour=Codon,label=Codon)) + 
    geom_text() + 
    geom_line(aes(group=Codon)) + 
    scale_y_continuous(limits=c(0,0.1),expand=c(0,0)) +
    theme(legend.position="none")
```


### Does depletion differ depending on the frame?

```{r ccposframe,dependson="load_cds",results="show"}
ccposframe <- bind_rows(
    CDS %>% 
        subseq(start = 1) %>%
        countCodonsPos %>%
        mutate(Frame=0),
        CDS %>% 
        subseq(start = 2) %>%
        countCodonsPos %>%
        mutate(Frame=1),
        CDS %>% 
        subseq(start = 3) %>%
        countCodonsPos %>%
        mutate(Frame=2) ) %>%
    ungroup %>%
    mutate(Frame=factor(Frame))
ccposframe
```

### Depletion of ATG codons is wildly different depending on frame

#### Not smoothed

```{r plot_ccposframe_ATG,dependson="ccposframe",fig.height=3,fig.width=5}
ggplot(data=ccposframe %>% filter(Codon=="ATG"),
       aes(x=Pos,y=Freq,colour=Frame)) +
    geom_line(aes(group=Frame),size=0.7) + 
    scale_x_continuous("Position",limits=c(2,400),expand=c(0,1)) + 
    scale_y_continuous("ATG Frequency",limits=c(0,0.04),expand=c(0,0)) +
    scale_colour_brewer(palette="Set2")
```

#### Smoothed

```{r plot_ccposframe_smooth_ATG,dependson="ccposframe",fig.height=3,fig.width=5}
ggplot(data=ccposframe %>% filter(Codon=="ATG"),
       aes(x=Pos,y=Freq,colour=Frame)) +
    geom_smooth(aes(group=Frame),size=0.9,
                se=FALSE,span=0.05,method="loess") + 
    scale_x_continuous("Position",limits=c(2,400),expand=c(0,1)) + 
    scale_y_continuous("ATG Frequency",limits=c(0,0.04),expand=c(0,0)) +
    scale_colour_brewer(palette="Set2")
```

### Depletion/enrichment of many codons is wildly different depending on frame

Also *not* smoothed.

```{r plot_ccposframe_all,dependson="ccposframe",fig.width=7,fig.height=10}
ggplot(data=ccposframe %>% filter(Codon %in% names(GENETIC_CODE)),
       aes(x=Pos,y=Freq,colour=Frame)) +
    geom_line(aes(group=Frame),size=0.5) +
    # geom_smooth(aes(group=Frame),colour="grey50",
    #             size=0.4,se=FALSE,span=0.1,method="loess") + 
    scale_x_continuous("Position",limits=c(2,400),expand=c(0,1)) + 
    scale_y_continuous("Codon Frequency",limits=c(0,0.05),expand=c(0,0),
                       breaks=c(0,0.02,0.04)) +
    scale_colour_brewer(palette="Set2") +
    facet_wrap(~Codon,ncol=4) +
    theme(legend.position="top", 
          panel.border = element_rect(colour = "grey80",size = 0.5,
                                      fill=NA,linetype=1) )
```

ATG and indeed NTG codons are strongly depleted in frame 2, consistent with TGN codons depleted in frame 0 due to avoiding premature termination and the rare amino acids Cysteine and Tryptophan.

ATG and some other ATN codons are depleted in frame 1. Most codons NNG are depleted near the start in frame 1, as are many GNN in frame 0 (GAA, GAG, GAT, GGA,GGG,GGT). In fact G-rich codons seem to be depleted near-start relative to C-rich codons.

There is so much data here it is difficult to judge.

### Normalized in-frame to out-of-frame ratio, per Malabat et al.

Malabat et al compare change of in-frame to out-frame depletion by

For each codon:

1. Taking a ratio of total in:out of frame counts for codons 500 to 1000.
2. Taking a ratio of in:out of frame counts over 9-codon windows
3. Normalizing the latter by the former in log2-space.
4. Plot the smoothed normalized log2-ratios for each codon as function of position.
5. Comparing this for genes with and without internal transcription start sites (iTSS).

EW implemented this, using fixed-width 10-codon windows instead of the smoothed windows from Malabat et al.

```{r Mframeratios,dependson="ccposframe",fig.height=3,fig.width=5,results="show"}
ccnormratios <- ccposframe %>%
    filter(Pos > 500, Pos <= 1000) %>%
    select(-Pos) %>%
    group_by(Codon) %>%
    summarize(Frame0=sum(n*(Frame==0),na.rm=T),
              Frame1=sum(n*(Frame==1),na.rm=T),
              Frame2=sum(n*(Frame==2),na.rm=T),
              pInFrame=Frame0/(Frame1+Frame2))

ccbinratios <- ccposframe %>%
    filter(Pos <= 500, Pos >= 2, Codon %in% sensecodons) %>%
    mutate(PosCut = cut(Pos,breaks=seq(0,500,10),labels=seq(5,495,10))) %>%
    group_by(Codon,PosCut) %>%
    summarize(Frame0=sum(n*(Frame==0),na.rm=T),
              Frame1=sum(n*(Frame==1),na.rm=T),
              Frame2=sum(n*(Frame==2),na.rm=T),
              pInFrame=Frame0/(Frame1+Frame2)) %>%
    left_join( ccnormratios %>% select(Codon,pInNorm=pInFrame) ) %>%
    mutate(InFrameNorm=pInFrame/pInNorm,
           PosCut=PosCut %>% as.character %>% as.numeric  )

ggplot(data=ccbinratios, aes(x=PosCut,y=InFrameNorm) ) +
    geom_line(aes(group=Codon),colour="grey50",
              size=0.4) + 
    geom_line(data=ccbinratios %>% filter(Codon == "ATG"),
              colour="red",size=1.2) +
    scale_x_continuous(expand=c(0,1)) +
    labs(x="Position, Codons from aATG", y="In:Out of frame\n normalized ratio")
```


### Are good-context ATGs more depleted?

We need to count ATGs with good vs bad context. Maybe just a -3 A would work to give initial picture? Plot levels of TNNATG, CNNATG, ANNATG, GNNATG?

```{r ATGcontextbase,dependson="load_cds",results="show"}
sixmerATGlevels <- 
CDS %>%
    head %>%
    groupThingPos(width=6L,variable="sixmer") %>%
    count(sixmer)
sixmerATGlevels
```

